diff --git a/src/backend/commands/tablecmds.c b/src/backend/commands/tablecmds.c
index 0cd63f4e..fb64730a 100644
--- a/src/backend/commands/tablecmds.c
+++ b/src/backend/commands/tablecmds.c
@@ -19144,14 +19144,13 @@ ATExecDetachPartition(List **wqueue, AlteredTableInfo *tab, Relation rel,
 	Relation	partRel;
 	ObjectAddress address;
 	Oid			defaultPartOid;
-	PartitionDesc partdesc;
 
 	/*
 	 * We must lock the default partition, because detaching this partition
 	 * will change its partition constraint.
 	 */
-	partdesc = RelationGetPartitionDesc(rel, true);
-	defaultPartOid = get_default_oid_from_partdesc(partdesc);
+	defaultPartOid =
+		get_default_oid_from_partdesc(RelationGetPartitionDesc(rel, true));
 	if (OidIsValid(defaultPartOid))
 	{
 		/*
@@ -19218,13 +19217,10 @@ ATExecDetachPartition(List **wqueue, AlteredTableInfo *tab, Relation rel,
 		char	   *partrelname;
 
 		/*
-		 * For strategies other than hash, add a constraint to the partition
-		 * being detached which supplants the partition constraint. For hash
-		 * we cannot do that, because the constraint would reference the
-		 * partitioned table OID, possibly causing problems later.
+		 * Add a new constraint to the partition being detached, which
+		 * supplants the partition constraint (unless there is one already).
 		 */
-		if (partdesc->boundinfo->strategy != PARTITION_STRATEGY_HASH)
-			DetachAddConstraintIfNeeded(wqueue, partRel);
+		DetachAddConstraintIfNeeded(wqueue, partRel);
 
 		/*
 		 * We're almost done now; the only traces that remain are the
