diff --git a/src/backend/parser/parse_agg.c b/src/backend/parser/parse_agg.c
index be88909d..b15d9b15 100644
--- a/src/backend/parser/parse_agg.c
+++ b/src/backend/parser/parse_agg.c
@@ -1961,7 +1961,7 @@ resolve_aggregate_transtype(Oid aggfuncid,
 
 /*
  * agg_args_support_sendreceive
- *		Returns true if all non-byval types of aggref's args have send and
+ *		Returns true if all non-byval of aggref's arg types have send and
  *		receive functions.
  */
 bool
@@ -1976,15 +1976,6 @@ agg_args_support_sendreceive(Aggref *aggref)
 		TargetEntry *tle = (TargetEntry *) lfirst(lc);
 		Oid			type = exprType((Node *) tle->expr);
 
-		/*
-		 * RECORD is a special case: it has typsend/typreceive functions, but
-		 * record_recv only works if passed the correct typmod to identify the
-		 * specific anonymous record type.  array_agg_deserialize cannot do
-		 * that, so we have to disclaim support for the case.
-		 */
-		if (type == RECORDOID)
-			return false;
-
 		typeTuple = SearchSysCache1(TYPEOID, ObjectIdGetDatum(type));
 		if (!HeapTupleIsValid(typeTuple))
 			elog(ERROR, "cache lookup failed for type %u", type);
