diff --git a/src/backend/commands/copyto.c b/src/backend/commands/copyto.c
index 84dc465cb..eb1d3d8fb 100644
--- a/src/backend/commands/copyto.c
+++ b/src/backend/commands/copyto.c
@@ -475,7 +475,7 @@ BeginCopyTo(ParseState *pstate,
 				if (q->querySource == QSRC_NON_INSTEAD_RULE)
 					ereport(ERROR,
 							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-							 errmsg("DO ALSO rules are not supported for COPY")));
+							 errmsg("DO ALSO rules are not supported for the COPY")));
 			}
 
 			ereport(ERROR,
@@ -492,11 +492,7 @@ BeginCopyTo(ParseState *pstate,
 					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 					 errmsg("COPY (SELECT INTO) is not supported")));
 
-		/* The only other utility command we could see is NOTIFY */
-		if (query->utilityStmt != NULL)
-			ereport(ERROR,
-					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					 errmsg("COPY query must not be a utility command")));
+		Assert(query->utilityStmt == NULL);
 
 		/*
 		 * Similarly the grammar doesn't enforce the presence of a RETURNING
