diff --git a/src/backend/commands/functioncmds.c b/src/backend/commands/functioncmds.c
index 28c38fdb..d43b89d3 100644
--- a/src/backend/commands/functioncmds.c
+++ b/src/backend/commands/functioncmds.c
@@ -91,32 +91,11 @@ compute_return_type(TypeName *returnType, Oid languageOid,
 	Oid			rettype;
 	Type		typtup;
 	AclResult	aclresult;
-	bool		attempt_shell_creation;
-
-	/*
-	 * If this looks like it could be an input function, and the type doesn't
-	 * exist, we'll create it as a shell type.
-	 *
-	 * If the type name contains any modifiers like %TYPE, type[] array
-	 * syntax, or typmod decoration, it's not an input function, or at least
-	 * not one for which we'd want to automatically create a shell type.
-	 *
-	 * Only C-coded functions can be I/O functions.  We enforce this
-	 * restriction here mainly to prevent littering the catalogs with shell
-	 * types due to simple typos in user-defined function definitions.
-	 */
-	attempt_shell_creation =
-		!returnType->pct_type && returnType->arrayBounds == NULL &&
-		returnType->typmods == NIL &&
-		(languageOid == INTERNALlanguageId || languageOid == ClanguageId);
 
 	typtup = LookupTypeName(NULL, returnType, NULL, false);
+
 	if (typtup)
 	{
-		/*
-		 * Found an existing type with the given name.  Check if it's a shell
-		 * type.
-		 */
 		if (!((Form_pg_type) GETSTRUCT(typtup))->typisdefined)
 		{
 			if (languageOid == SQLlanguageId)
@@ -133,27 +112,37 @@ compute_return_type(TypeName *returnType, Oid languageOid,
 		rettype = typeTypeId(typtup);
 		ReleaseSysCache(typtup);
 	}
-	else if (!attempt_shell_creation)
-	{
-		/* Type not found and we don't want to create a shell type */
-		ereport(ERROR,
-				(errcode(ERRCODE_UNDEFINED_OBJECT),
-				 errmsg("type \"%s\" does not exist",
-						TypeNameToString(returnType))));
-	}
 	else
 	{
-		/* Make a shell type */
+		char	   *typnam = TypeNameToString(returnType);
 		Oid			namespaceId;
 		char	   *typname;
 		ObjectAddress address;
 
+		/*
+		 * Only C-coded functions can be I/O functions.  We enforce this
+		 * restriction here mainly to prevent littering the catalogs with
+		 * shell types due to simple typos in user-defined function
+		 * definitions.
+		 */
+		if (languageOid != INTERNALlanguageId &&
+			languageOid != ClanguageId)
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_OBJECT),
+					 errmsg("type \"%s\" does not exist", typnam)));
+
+		/* Reject if there's typmod decoration, too */
+		if (returnType->typmods != NIL)
+			ereport(ERROR,
+					(errcode(ERRCODE_SYNTAX_ERROR),
+					 errmsg("type modifier cannot be specified for shell type \"%s\"",
+							typnam)));
+
+		/* Otherwise, go ahead and make a shell type */
 		ereport(NOTICE,
 				(errcode(ERRCODE_UNDEFINED_OBJECT),
-				 errmsg("type \"%s\" is not yet defined",
-						TypeNameToString(returnType)),
+				 errmsg("type \"%s\" is not yet defined", typnam),
 				 errdetail("Creating a shell type definition.")));
-
 		namespaceId = QualifiedNameGetCreationNamespace(returnType->names,
 														&typname);
 		aclresult = object_aclcheck(NamespaceRelationId, namespaceId, GetUserId(),
@@ -161,7 +150,6 @@ compute_return_type(TypeName *returnType, Oid languageOid,
 		if (aclresult != ACLCHECK_OK)
 			aclcheck_error(aclresult, OBJECT_SCHEMA,
 						   get_namespace_name(namespaceId));
-
 		address = TypeShellMake(typname, namespaceId, GetUserId());
 		rettype = address.objectId;
 		Assert(OidIsValid(rettype));
