diff --git a/src/backend/utils/adt/varlena.c b/src/backend/utils/adt/varlena.c
index 616eff67..853c1459 100644
--- a/src/backend/utils/adt/varlena.c
+++ b/src/backend/utils/adt/varlena.c
@@ -130,7 +130,6 @@ static text *text_substring(Datum str,
 							int32 start,
 							int32 length,
 							bool length_not_specified);
-static int	pg_mbcharcliplen_chars(const char *mbstr, int len, int limit);
 static text *text_overlay(text *t1, text *t2, int sp, int sl);
 static int	text_position(text *t1, text *t2, Oid collid);
 static void text_position_setup(text *t1, text *t2, Oid collid, TextPositionState *state);
@@ -892,7 +891,7 @@ text_substring(Datum str, int32 start, int32 length, bool length_not_specified)
 	int32		S = start;		/* start position */
 	int32		S1;				/* adjusted start position */
 	int32		L1;				/* adjusted substring length */
-	int32		E;				/* end position, exclusive */
+	int32		E;				/* end position */
 
 	/*
 	 * SQL99 says S can be zero or negative (which we don't document), but we
@@ -990,11 +989,11 @@ text_substring(Datum str, int32 start, int32 length, bool length_not_specified)
 		else
 		{
 			/*
-			 * Ending at position 1, exclusive, obviously yields an empty
-			 * string.  A zero or negative value can happen if the start was
-			 * negative or one. SQL99 says to return a zero-length string.
+			 * A zero or negative value for the end position can happen if the
+			 * start was negative or one. SQL99 says to return a zero-length
+			 * string.
 			 */
-			if (E <= 1)
+			if (E < 1)
 				return cstring_to_text("");
 
 			/*
@@ -1004,11 +1003,11 @@ text_substring(Datum str, int32 start, int32 length, bool length_not_specified)
 			L1 = E - S1;
 
 			/*
-			 * Total slice size in bytes can't be any longer than the
-			 * inclusive end position times the encoding max length.  If that
-			 * overflows, we can just use -1.
+			 * Total slice size in bytes can't be any longer than the start
+			 * position plus substring length times the encoding max length.
+			 * If that overflows, we can just use -1.
 			 */
-			if (pg_mul_s32_overflow(E - 1, eml, &slice_size))
+			if (pg_mul_s32_overflow(E, eml, &slice_size))
 				slice_size = -1;
 		}
 
@@ -1031,17 +1030,9 @@ text_substring(Datum str, int32 start, int32 length, bool length_not_specified)
 			return cstring_to_text("");
 		}
 
-		/*
-		 * Now we can get the actual length of the slice in MB characters,
-		 * stopping at the end of the substring.  Continuing beyond the
-		 * substring end could find an incomplete character attributable
-		 * solely to DatumGetTextPSlice() chopping in the middle of a
-		 * character, and it would be superfluous work at best.
-		 */
-		slice_strlen =
-			(slice_size == -1 ?
-			 pg_mbstrlen_with_len(VARDATA_ANY(slice), slice_len) :
-			 pg_mbcharcliplen_chars(VARDATA_ANY(slice), slice_len, E - 1));
+		/* Now we can get the actual length of the slice in MB characters */
+		slice_strlen = pg_mbstrlen_with_len(VARDATA_ANY(slice),
+											slice_len);
 
 		/*
 		 * Check that the start position wasn't > slice_strlen. If so, SQL99
@@ -1096,35 +1087,6 @@ text_substring(Datum str, int32 start, int32 length, bool length_not_specified)
 	return NULL;
 }
 
-/*
- * pg_mbcharcliplen_chars -
- *	Mirror pg_mbcharcliplen(), except return value unit is chars, not bytes.
- *
- *	This mirrors all the dubious historical behavior, so it's static to
- *	discourage proliferation.  The assertions are specific to the one caller.
- */
-static int
-pg_mbcharcliplen_chars(const char *mbstr, int len, int limit)
-{
-	int			nch = 0;
-	int			l;
-
-	Assert(len > 0);
-	Assert(limit > 0);
-	Assert(pg_database_encoding_max_length() > 1);
-
-	while (len > 0 && *mbstr)
-	{
-		l = pg_mblen_with_len(mbstr, len);
-		nch++;
-		if (nch == limit)
-			break;
-		len -= l;
-		mbstr += l;
-	}
-	return nch;
-}
-
 /*
  * textoverlay
  *	Replace specified substring of first string with second
