diff --git a/src/backend/commands/tablecmds.c b/src/backend/commands/tablecmds.c
index 3f508760..36717ffc 100644
--- a/src/backend/commands/tablecmds.c
+++ b/src/backend/commands/tablecmds.c
@@ -19258,7 +19258,6 @@ DetachPartitionFinalize(Relation rel, Relation partRel, bool concurrent,
 	HeapTuple	tuple,
 				newtuple;
 	Relation	trigrel = NULL;
-	List	   *fkoids = NIL;
 
 	if (concurrent)
 	{
@@ -19279,23 +19278,6 @@ DetachPartitionFinalize(Relation rel, Relation partRel, bool concurrent,
 	fks = copyObject(RelationGetFKeyList(partRel));
 	if (fks != NIL)
 		trigrel = table_open(TriggerRelationId, RowExclusiveLock);
-
-	/*
-	 * It's possible that the partition being detached has a foreign key that
-	 * references a partitioned table.  When that happens, there are multiple
-	 * pg_constraint rows for the partition: one points to the partitioned
-	 * table itself, while the others point to each of its partitions.  Only
-	 * the topmost one is to be considered here; the child constraints must be
-	 * left alone, because conceptually those aren't coming from our parent
-	 * partitioned table, but from this partition itself.
-	 *
-	 * We implement this by collecting all the constraint OIDs in a first scan
-	 * of the FK array, and skipping in the loop below those constraints whose
-	 * parents are listed here.
-	 */
-	foreach_node(ForeignKeyCacheInfo, fk, fks)
-		fkoids = lappend_oid(fkoids, fk->conoid);
-
 	foreach(cell, fks)
 	{
 		ForeignKeyCacheInfo *fk = lfirst(cell);
@@ -19309,13 +19291,9 @@ DetachPartitionFinalize(Relation rel, Relation partRel, bool concurrent,
 			elog(ERROR, "cache lookup failed for constraint %u", fk->conoid);
 		conform = (Form_pg_constraint) GETSTRUCT(contup);
 
-		/*
-		 * Consider only inherited foreign keys, and only if their parents
-		 * aren't in the list.
-		 */
+		/* consider only the inherited foreign keys */
 		if (conform->contype != CONSTRAINT_FOREIGN ||
-			!OidIsValid(conform->conparentid) ||
-			list_member_oid(fkoids, conform->conparentid))
+			!OidIsValid(conform->conparentid))
 		{
 			ReleaseSysCache(contup);
 			continue;
