diff --git a/src/backend/executor/nodeIndexonlyscan.c b/src/backend/executor/nodeIndexonlyscan.c
index b12a7b1c..b49194c0 100644
--- a/src/backend/executor/nodeIndexonlyscan.c
+++ b/src/backend/executor/nodeIndexonlyscan.c
@@ -31,7 +31,6 @@
 #include "postgres.h"
 
 #include "access/genam.h"
-#include "access/htup_details.h"
 #include "access/relscan.h"
 #include "access/tableam.h"
 #include "access/tupdesc.h"
@@ -49,7 +48,7 @@
 
 static TupleTableSlot *IndexOnlyNext(IndexOnlyScanState *node);
 static void StoreIndexTuple(IndexOnlyScanState *node, TupleTableSlot *slot,
-							IndexScanDesc scandesc);
+							IndexTuple itup, TupleDesc itupdesc);
 
 
 /* ----------------------------------------------------------------
@@ -190,8 +189,27 @@ IndexOnlyNext(IndexOnlyScanState *node)
 			tuple_from_heap = true;
 		}
 
-		/* Fill the scan tuple slot with data from the index */
-		StoreIndexTuple(node, slot, scandesc);
+		/*
+		 * Fill the scan tuple slot with data from the index.  This might be
+		 * provided in either HeapTuple or IndexTuple format.  Conceivably an
+		 * index AM might fill both fields, in which case we prefer the heap
+		 * format, since it's probably a bit cheaper to fill a slot from.
+		 */
+		if (scandesc->xs_hitup)
+		{
+			/*
+			 * We don't take the trouble to verify that the provided tuple has
+			 * exactly the slot's format, but it seems worth doing a quick
+			 * check on the number of fields.
+			 */
+			Assert(slot->tts_tupleDescriptor->natts ==
+				   scandesc->xs_hitupdesc->natts);
+			ExecForceStoreHeapTuple(scandesc->xs_hitup, slot, false);
+		}
+		else if (scandesc->xs_itup)
+			StoreIndexTuple(node, slot, scandesc->xs_itup, scandesc->xs_itupdesc);
+		else
+			elog(ERROR, "no data returned for index-only scan");
 
 		/*
 		 * If the index was lossy, we have to recheck the index quals.
@@ -241,79 +259,56 @@ IndexOnlyNext(IndexOnlyScanState *node)
 
 /*
  * StoreIndexTuple
- *		Fill the slot with the data the index AM returned.
- *
- * The data might be provided in either HeapTuple (xs_hitup) or IndexTuple
- * (xs_itup) format.  Conceivably an index AM might fill both fields, in which
- * case we prefer the heap format, since it's probably a bit cheaper to fill a
- * slot from.
+ *		Fill the slot with data from the index tuple.
  *
  * At some point this might be generally-useful functionality, but
  * right now we don't need it elsewhere.
  */
 static void
 StoreIndexTuple(IndexOnlyScanState *node, TupleTableSlot *slot,
-				IndexScanDesc scandesc)
+				IndexTuple itup, TupleDesc itupdesc)
 {
+	/*
+	 * Note: we must use the tupdesc supplied by the AM in index_deform_tuple,
+	 * not the slot's tupdesc, in case the latter has different datatypes
+	 * (this happens for btree name_ops in particular).  They'd better have
+	 * the same number of columns though, as well as being datatype-compatible
+	 * which is something we can't so easily check.
+	 */
+	Assert(slot->tts_tupleDescriptor->natts == itupdesc->natts);
+
 	ExecClearTuple(slot);
+	index_deform_tuple(itup, itupdesc, slot->tts_values, slot->tts_isnull);
 
 	/*
-	 * We must deform the tuple using the tupdesc the index AM formed it with
-	 * (xs_hitupdesc or xs_itupdesc), not the slot's tupdesc.  The datums
-	 * returned by the index AM must be binary compatible, but the descriptors
-	 * may align each column differently in certain rare cases. (Actually,
-	 * btree's "name" opclass stores cstring tuples that _aren't_ even binary
-	 * compatible, in the strictest sense.  We directly handle that here.)
+	 * Copy all name columns stored as cstrings back into a NAMEDATALEN byte
+	 * sized allocation.  We mark this branch as unlikely as generally "name"
+	 * is used only for the system catalogs and this would have to be a user
+	 * query running on those or some other user table with an index on a name
+	 * column.
 	 */
-	if (scandesc->xs_hitup)
+	if (unlikely(node->ioss_NameCStringAttNums != NULL))
 	{
-		Assert(slot->tts_tupleDescriptor->natts == scandesc->xs_hitupdesc->natts);
+		int			attcount = node->ioss_NameCStringCount;
 
-		heap_deform_tuple(scandesc->xs_hitup, scandesc->xs_hitupdesc,
-						  slot->tts_values, slot->tts_isnull);
-	}
-	else if (scandesc->xs_itup)
-	{
-		Assert(slot->tts_tupleDescriptor->natts == scandesc->xs_itupdesc->natts);
+		for (int idx = 0; idx < attcount; idx++)
+		{
+			int			attnum = node->ioss_NameCStringAttNums[idx];
+			Name		name;
 
-		index_deform_tuple(scandesc->xs_itup, scandesc->xs_itupdesc,
-						   slot->tts_values, slot->tts_isnull);
+			/* skip null Datums */
+			if (slot->tts_isnull[attnum])
+				continue;
 
-		/*
-		 * Copy all name columns stored as cstrings back into a NAMEDATALEN
-		 * byte sized allocation.  We mark this branch as unlikely as
-		 * generally "name" is used only for the system catalogs and this
-		 * would have to be a user query running on those or some other user
-		 * table with an index on a name column.
-		 */
-		if (unlikely(node->ioss_NameCStringAttNums != NULL))
-		{
-			int			attcount = node->ioss_NameCStringCount;
+			/* allocate the NAMEDATALEN and copy the datum into that memory */
+			name = (Name) MemoryContextAlloc(node->ss.ps.ps_ExprContext->ecxt_per_tuple_memory,
+											 NAMEDATALEN);
 
-			for (int idx = 0; idx < attcount; idx++)
-			{
-				int			attnum = node->ioss_NameCStringAttNums[idx];
-				Name		name;
-
-				/* skip null Datums */
-				if (slot->tts_isnull[attnum])
-					continue;
-
-				/*
-				 * allocate the NAMEDATALEN and copy the datum into that
-				 * memory
-				 */
-				name = (Name) MemoryContextAlloc(node->ss.ps.ps_ExprContext->ecxt_per_tuple_memory,
-												 NAMEDATALEN);
-
-				/* use namestrcpy to zero-pad all trailing bytes */
-				namestrcpy(name, DatumGetCString(slot->tts_values[attnum]));
-				slot->tts_values[attnum] = NameGetDatum(name);
-			}
+			/* use namestrcpy to zero-pad all trailing bytes */
+			namestrcpy(name, DatumGetCString(slot->tts_values[attnum]));
+			slot->tts_values[attnum] = NameGetDatum(name);
 		}
 	}
-	else
-		elog(ERROR, "no data returned for index-only scan");
 
 	ExecStoreVirtualTuple(slot);
 }
