diff --git a/src/backend/commands/tablecmds.c b/src/backend/commands/tablecmds.c
index a83a4a0a1..3ba65a33a 100644
--- a/src/backend/commands/tablecmds.c
+++ b/src/backend/commands/tablecmds.c
@@ -19899,10 +19899,7 @@ ATExecAttachPartitionIdx(List **wqueue, Relation parentIdx, RangeVar *name)
 
 	ObjectAddressSet(address, RelationRelationId, RelationGetRelid(partIdx));
 
-	/*
-	 * Check if the index is already attached to the correct parent,
-	 * ultimately attempting one round of validation if already the case.
-	 */
+	/* Silently do nothing if already in the right state */
 	currParent = partIdx->rd_rel->relispartition ?
 		get_partition_parent(partIdxId, false) : InvalidOid;
 	if (currParent != RelationGetRelid(parentIdx))
@@ -20004,14 +20001,6 @@ ATExecAttachPartitionIdx(List **wqueue, Relation parentIdx, RangeVar *name)
 
 		validatePartitionedIndex(parentIdx, parentTbl);
 	}
-	else if (!parentIdx->rd_index->indisvalid)
-	{
-		/*
-		 * The index is attached, but the parent is still invalid; see if it
-		 * can be validated now.
-		 */
-		validatePartitionedIndex(parentIdx, parentTbl);
-	}
 
 	relation_close(parentTbl, AccessShareLock);
 	/* keep these locks till commit */
