diff --git a/src/backend/utils/adt/selfuncs.c b/src/backend/utils/adt/selfuncs.c
index f4205179..f4b3e91b 100644
--- a/src/backend/utils/adt/selfuncs.c
+++ b/src/backend/utils/adt/selfuncs.c
@@ -121,7 +121,6 @@
 #include "parser/parse_clause.h"
 #include "parser/parse_relation.h"
 #include "parser/parsetree.h"
-#include "rewrite/rewriteManip.h"
 #include "statistics/statistics.h"
 #include "storage/bufmgr.h"
 #include "utils/acl.h"
@@ -3307,15 +3306,6 @@ add_unique_group_var(PlannerInfo *root, List *varinfos,
 
 	ndistinct = get_variable_numdistinct(vardata, &isdefault);
 
-	/*
-	 * The nullingrels bits within the var could cause the same var to be
-	 * counted multiple times if it's marked with different nullingrels.  They
-	 * could also prevent us from matching the var to the expressions in
-	 * extended statistics (see estimate_multivariate_ndistinct).  So strip
-	 * them out first.
-	 */
-	var = remove_nulling_relids(var, root->outer_join_rels, NULL);
-
 	foreach(lc, varinfos)
 	{
 		varinfo = (GroupVarInfo *) lfirst(lc);
@@ -5027,7 +5017,6 @@ examine_variable(PlannerInfo *root, Node *node, int varRelid,
 {
 	Node	   *basenode;
 	Relids		varnos;
-	Relids		basevarnos;
 	RelOptInfo *onerel;
 
 	/* Make sure we don't return dangling pointers in vardata */
@@ -5069,11 +5058,10 @@ examine_variable(PlannerInfo *root, Node *node, int varRelid,
 	 * relation are considered "real" vars.
 	 */
 	varnos = pull_varnos(root, basenode);
-	basevarnos = bms_difference(varnos, root->outer_join_rels);
 
 	onerel = NULL;
 
-	if (bms_is_empty(basevarnos))
+	if (bms_is_empty(varnos))
 	{
 		/* No Vars at all ... must be pseudo-constant clause */
 	}
@@ -5081,8 +5069,7 @@ examine_variable(PlannerInfo *root, Node *node, int varRelid,
 	{
 		int			relid;
 
-		/* Check if the expression is in vars of a single base relation */
-		if (bms_get_singleton_member(basevarnos, &relid))
+		if (bms_get_singleton_member(varnos, &relid))
 		{
 			if (varRelid == 0 || varRelid == relid)
 			{
@@ -5112,7 +5099,7 @@ examine_variable(PlannerInfo *root, Node *node, int varRelid,
 		}
 	}
 
-	bms_free(basevarnos);
+	bms_free(varnos);
 
 	vardata->var = node;
 	vardata->atttype = exprType(node);
@@ -5137,14 +5124,6 @@ examine_variable(PlannerInfo *root, Node *node, int varRelid,
 		ListCell   *slist;
 		Oid			userid;
 
-		/*
-		 * The nullingrels bits within the expression could prevent us from
-		 * matching it to expressional index columns or to the expressions in
-		 * extended statistics.  So strip them out first.
-		 */
-		if (bms_overlap(varnos, root->outer_join_rels))
-			node = remove_nulling_relids(node, root->outer_join_rels, NULL);
-
 		/*
 		 * Determine the user ID to use for privilege checks: either
 		 * onerel->userid if it's set (e.g., in case we're accessing the table
@@ -5415,8 +5394,6 @@ examine_variable(PlannerInfo *root, Node *node, int varRelid,
 			}
 		}
 	}
-
-	bms_free(varnos);
 }
 
 /*
