diff --git a/src/backend/catalog/heap.c b/src/backend/catalog/heap.c
index 3780ed46a..2d66639a2 100644
--- a/src/backend/catalog/heap.c
+++ b/src/backend/catalog/heap.c
@@ -648,16 +648,6 @@ CheckAttributeType(const char *attname,
 						   containing_rowtypes,
 						   flags);
 	}
-	else if (att_typtype == TYPTYPE_MULTIRANGE)
-	{
-		/*
-		 * If it's a multirange, recurse to check its plain range type.
-		 */
-		CheckAttributeType(attname, get_multirange_range(atttypid),
-						   InvalidOid,	/* range types are not collatable */
-						   containing_rowtypes,
-						   flags);
-	}
 	else if (OidIsValid((att_typelem = get_element_type(atttypid))))
 	{
 		/*
