diff --git a/src/backend/executor/nodeModifyTable.c b/src/backend/executor/nodeModifyTable.c
index a0d1091e..d499f842 100644
--- a/src/backend/executor/nodeModifyTable.c
+++ b/src/backend/executor/nodeModifyTable.c
@@ -64,7 +64,6 @@
 #include "nodes/nodeFuncs.h"
 #include "optimizer/optimizer.h"
 #include "rewrite/rewriteHandler.h"
-#include "rewrite/rewriteManip.h"
 #include "storage/lmgr.h"
 #include "utils/builtins.h"
 #include "utils/datum.h"
@@ -3552,7 +3551,6 @@ ExecInitMerge(ModifyTableState *mtstate, EState *estate)
 			switch (action->commandType)
 			{
 				case CMD_INSERT:
-					/* INSERT actions always use rootRelInfo */
 					ExecCheckPlanOutput(rootRelInfo->ri_RelationDesc,
 										action->targetList);
 
@@ -3592,23 +3590,9 @@ ExecInitMerge(ModifyTableState *mtstate, EState *estate)
 					}
 					else
 					{
-						/*
-						 * If the MERGE targets an inherited table, we insert
-						 * into the root table, so we must initialize its
-						 * "new" tuple slot, if not already done, and use its
-						 * relation descriptor for the projection.
-						 *
-						 * For non-inherited tables, rootRelInfo and
-						 * resultRelInfo are the same, and the "new" tuple
-						 * slot will already have been initialized.
-						 */
-						if (rootRelInfo->ri_newTupleSlot == NULL)
-							rootRelInfo->ri_newTupleSlot =
-								table_slot_create(rootRelInfo->ri_RelationDesc,
-												  &estate->es_tupleTable);
-
-						tgtslot = rootRelInfo->ri_newTupleSlot;
-						tgtdesc = RelationGetDescr(rootRelInfo->ri_RelationDesc);
+						/* not partitioned? use the stock relation and slot */
+						tgtslot = resultRelInfo->ri_newTupleSlot;
+						tgtdesc = RelationGetDescr(resultRelInfo->ri_RelationDesc);
 					}
 
 					action_state->mas_proj =
@@ -3641,113 +3625,6 @@ ExecInitMerge(ModifyTableState *mtstate, EState *estate)
 			}
 		}
 	}
-
-	/*
-	 * If the MERGE targets an inherited table, any INSERT actions will use
-	 * rootRelInfo, and rootRelInfo will not be in the resultRelInfo array.
-	 * Therefore we must initialize its WITH CHECK OPTION constraints and
-	 * RETURNING projection, as ExecInitModifyTable did for the resultRelInfo
-	 * entries.
-	 *
-	 * Note that the planner does not build a withCheckOptionList or
-	 * returningList for the root relation, but as in ExecInitPartitionInfo,
-	 * we can use the first resultRelInfo entry as a reference to calculate
-	 * the attno's for the root table.
-	 */
-	if (rootRelInfo != mtstate->resultRelInfo &&
-		rootRelInfo->ri_RelationDesc->rd_rel->relkind != RELKIND_PARTITIONED_TABLE &&
-		(mtstate->mt_merge_subcommands & MERGE_INSERT) != 0)
-	{
-		Relation	rootRelation = rootRelInfo->ri_RelationDesc;
-		Relation	firstResultRel = mtstate->resultRelInfo[0].ri_RelationDesc;
-		int			firstVarno = mtstate->resultRelInfo[0].ri_RangeTableIndex;
-		AttrMap    *part_attmap = NULL;
-		bool		found_whole_row;
-
-		if (node->withCheckOptionLists != NIL)
-		{
-			List	   *wcoList;
-			List	   *wcoExprs = NIL;
-
-			/* There should be as many WCO lists as result rels */
-			Assert(list_length(node->withCheckOptionLists) ==
-				   list_length(node->resultRelations));
-
-			/*
-			 * Use the first WCO list as a reference. In the most common case,
-			 * this will be for the same relation as rootRelInfo, and so there
-			 * will be no need to adjust its attno's.
-			 */
-			wcoList = linitial(node->withCheckOptionLists);
-			if (rootRelation != firstResultRel)
-			{
-				/* Convert any Vars in it to contain the root's attno's */
-				part_attmap =
-					build_attrmap_by_name(RelationGetDescr(rootRelation),
-										  RelationGetDescr(firstResultRel),
-										  false);
-
-				wcoList = (List *)
-					map_variable_attnos((Node *) wcoList,
-										firstVarno, 0,
-										part_attmap,
-										RelationGetForm(rootRelation)->reltype,
-										&found_whole_row);
-			}
-
-			foreach(lc, wcoList)
-			{
-				WithCheckOption *wco = lfirst_node(WithCheckOption, lc);
-				ExprState  *wcoExpr = ExecInitQual(castNode(List, wco->qual),
-												   &mtstate->ps);
-
-				wcoExprs = lappend(wcoExprs, wcoExpr);
-			}
-
-			rootRelInfo->ri_WithCheckOptions = wcoList;
-			rootRelInfo->ri_WithCheckOptionExprs = wcoExprs;
-		}
-
-		if (node->returningLists != NIL)
-		{
-			List	   *returningList;
-
-			/* There should be as many returning lists as result rels */
-			Assert(list_length(node->returningLists) ==
-				   list_length(node->resultRelations));
-
-			/*
-			 * Use the first returning list as a reference. In the most common
-			 * case, this will be for the same relation as rootRelInfo, and so
-			 * there will be no need to adjust its attno's.
-			 */
-			returningList = linitial(node->returningLists);
-			if (rootRelation != firstResultRel)
-			{
-				/* Convert any Vars in it to contain the root's attno's */
-				if (part_attmap == NULL)
-					part_attmap =
-						build_attrmap_by_name(RelationGetDescr(rootRelation),
-											  RelationGetDescr(firstResultRel),
-											  false);
-
-				returningList = (List *)
-					map_variable_attnos((Node *) returningList,
-										firstVarno, 0,
-										part_attmap,
-										RelationGetForm(rootRelation)->reltype,
-										&found_whole_row);
-			}
-			rootRelInfo->ri_returningList = returningList;
-
-			/* Initialize the RETURNING projection */
-			rootRelInfo->ri_projectReturning =
-				ExecBuildProjectionInfo(returningList, econtext,
-										mtstate->ps.ps_ResultTupleSlot,
-										&mtstate->ps,
-										RelationGetDescr(rootRelation));
-		}
-	}
 }
 
 /*
