diff --git a/src/backend/optimizer/path/joinrels.c b/src/backend/optimizer/path/joinrels.c
index d4c5738e..db475e25 100644
--- a/src/backend/optimizer/path/joinrels.c
+++ b/src/backend/optimizer/path/joinrels.c
@@ -45,8 +45,7 @@ static void try_partitionwise_join(PlannerInfo *root, RelOptInfo *rel1,
 static SpecialJoinInfo *build_child_join_sjinfo(PlannerInfo *root,
 												SpecialJoinInfo *parent_sjinfo,
 												Relids left_relids, Relids right_relids);
-static void free_child_join_sjinfo(SpecialJoinInfo *child_sjinfo,
-								   SpecialJoinInfo *parent_sjinfo);
+static void free_child_join_sjinfo(SpecialJoinInfo *sjinfo);
 static void compute_partition_bounds(PlannerInfo *root, RelOptInfo *rel1,
 									 RelOptInfo *rel2, RelOptInfo *joinrel,
 									 SpecialJoinInfo *parent_sjinfo,
@@ -1678,7 +1677,7 @@ try_partitionwise_join(PlannerInfo *root, RelOptInfo *rel1, RelOptInfo *rel2,
 									child_restrictlist);
 
 		pfree(appinfos);
-		free_child_join_sjinfo(child_sjinfo, parent_sjinfo);
+		free_child_join_sjinfo(child_sjinfo);
 	}
 }
 
@@ -1745,33 +1744,18 @@ build_child_join_sjinfo(PlannerInfo *root, SpecialJoinInfo *parent_sjinfo,
  * SpecialJoinInfo are freed here.
  */
 static void
-free_child_join_sjinfo(SpecialJoinInfo *child_sjinfo,
-					   SpecialJoinInfo *parent_sjinfo)
+free_child_join_sjinfo(SpecialJoinInfo *sjinfo)
 {
 	/*
 	 * Dummy SpecialJoinInfos of inner joins do not have any translated fields
 	 * and hence no fields that to be freed.
 	 */
-	if (child_sjinfo->jointype != JOIN_INNER)
+	if (sjinfo->jointype != JOIN_INNER)
 	{
-		if (child_sjinfo->min_lefthand != parent_sjinfo->min_lefthand)
-			bms_free(child_sjinfo->min_lefthand);
-
-		if (child_sjinfo->min_righthand != parent_sjinfo->min_righthand)
-			bms_free(child_sjinfo->min_righthand);
-
-		if (child_sjinfo->syn_lefthand != parent_sjinfo->syn_lefthand)
-			bms_free(child_sjinfo->syn_lefthand);
-
-		if (child_sjinfo->syn_righthand != parent_sjinfo->syn_righthand)
-			bms_free(child_sjinfo->syn_righthand);
-
-		Assert(child_sjinfo->commute_above_l == parent_sjinfo->commute_above_l);
-		Assert(child_sjinfo->commute_above_r == parent_sjinfo->commute_above_r);
-		Assert(child_sjinfo->commute_below_l == parent_sjinfo->commute_below_l);
-		Assert(child_sjinfo->commute_below_r == parent_sjinfo->commute_below_r);
-
-		Assert(child_sjinfo->semi_operators == parent_sjinfo->semi_operators);
+		bms_free(sjinfo->min_lefthand);
+		bms_free(sjinfo->min_righthand);
+		bms_free(sjinfo->syn_lefthand);
+		bms_free(sjinfo->syn_righthand);
 
 		/*
 		 * semi_rhs_exprs may in principle be freed, but a simple pfree() does
@@ -1779,7 +1763,7 @@ free_child_join_sjinfo(SpecialJoinInfo *child_sjinfo,
 		 */
 	}
 
-	pfree(child_sjinfo);
+	pfree(sjinfo);
 }
 
 /*
