diff --git a/src/backend/partitioning/partprune.c b/src/backend/partitioning/partprune.c
index ea6bf2f1..88e31c24 100644
--- a/src/backend/partitioning/partprune.c
+++ b/src/backend/partitioning/partprune.c
@@ -3165,15 +3165,12 @@ get_matching_range_bounds(PartitionPruneContext *context,
 					 * of smallest such bound) or find the smallest one that's
 					 * greater than the lookup values and set minoff to that.
 					 */
-					while (true)
+					while (off >= 1 && off < boundinfo->ndatums - 1)
 					{
 						int32		cmpval;
 						int			nextoff;
 
 						nextoff = inclusive ? off - 1 : off + 1;
-
-						if (nextoff < 0 || nextoff >= boundinfo->ndatums)
-							break;
 						cmpval =
 							partition_rbound_datum_cmp(partsupfunc,
 													   partcollation,
@@ -3227,21 +3224,16 @@ get_matching_range_bounds(PartitionPruneContext *context,
 			if (off >= 0)
 			{
 				/*
-				 * As above, check adjacent bounds to see if the bound is
-				 * equal to the lookup value.
+				 * See the comment above.
 				 */
 				if (is_equal && nvalues < partnatts)
 				{
-					while (true)
+					while (off >= 1 && off < boundinfo->ndatums - 1)
 					{
 						int32		cmpval;
 						int			nextoff;
 
 						nextoff = inclusive ? off + 1 : off - 1;
-
-						if (nextoff < 0 || nextoff >= boundinfo->ndatums)
-							break;
-
 						cmpval = partition_rbound_datum_cmp(partsupfunc,
 															partcollation,
 															boundinfo->datums[nextoff],
