diff --git a/src/backend/utils/adt/acl.c b/src/backend/utils/adt/acl.c
index 6c723e0d..d7b39140 100644
--- a/src/backend/utils/adt/acl.c
+++ b/src/backend/utils/adt/acl.c
@@ -130,22 +130,6 @@ static AclResult pg_role_aclcheck(Oid role_oid, Oid roleid, AclMode mode);
 static void RoleMembershipCacheCallback(Datum arg, int cacheid, uint32 hashvalue);
 
 
-/*
- * Test whether an identifier char can be left unquoted in ACLs.
- *
- * Formerly, we used isalnum() even on non-ASCII characters, resulting in
- * unportable behavior.  To ensure dump compatibility with old versions,
- * we now treat high-bit-set characters as always requiring quoting during
- * putid(), but getid() will always accept them without quotes.
- */
-static inline bool
-is_safe_acl_char(unsigned char c, bool is_getid)
-{
-	if (IS_HIGHBIT_SET(c))
-		return is_getid;
-	return isalnum(c) || c == '_';
-}
-
 /*
  * getid
  *		Consumes the first alphanumeric string (identifier) found in string
@@ -171,22 +155,21 @@ getid(const char *s, char *n, Node *escontext)
 
 	while (isspace((unsigned char) *s))
 		s++;
+	/* This code had better match what putid() does, below */
 	for (;
 		 *s != '\0' &&
-		 (in_quotes || *s == '"' || is_safe_acl_char(*s, true));
+		 (isalnum((unsigned char) *s) ||
+		  *s == '_' ||
+		  *s == '"' ||
+		  in_quotes);
 		 s++)
 	{
 		if (*s == '"')
 		{
-			if (!in_quotes)
-			{
-				in_quotes = true;
-				continue;
-			}
 			/* safe to look at next char (could be '\0' though) */
 			if (*(s + 1) != '"')
 			{
-				in_quotes = false;
+				in_quotes = !in_quotes;
 				continue;
 			}
 			/* it's an escaped double quote; skip the escaping char */
@@ -220,10 +203,10 @@ putid(char *p, const char *s)
 	const char *src;
 	bool		safe = true;
 
-	/* Detect whether we need to use double quotes */
 	for (src = s; *src; src++)
 	{
-		if (!is_safe_acl_char(*src, false))
+		/* This test had better match what getid() does, above */
+		if (!isalnum((unsigned char) *src) && *src != '_')
 		{
 			safe = false;
 			break;
