diff --git a/src/backend/executor/execMain.c b/src/backend/executor/execMain.c
index d6129f5b..05f8c70d 100644
--- a/src/backend/executor/execMain.c
+++ b/src/backend/executor/execMain.c
@@ -1033,16 +1033,7 @@ CheckValidResultRelNew(ResultRelInfo *resultRelInfo, CmdType operation,
 	{
 		case RELKIND_RELATION:
 		case RELKIND_PARTITIONED_TABLE:
-
-			/*
-			 * For MERGE, check that the target relation supports each action.
-			 * For other operations, just check the operation itself.
-			 */
-			if (operation == CMD_MERGE)
-				foreach_node(MergeAction, action, mergeActions)
-					CheckCmdReplicaIdentity(resultRel, action->commandType);
-			else
-				CheckCmdReplicaIdentity(resultRel, operation);
+			CheckCmdReplicaIdentity(resultRel, operation);
 
 			/*
 			 * For INSERT ON CONFLICT DO UPDATE, additionally check that the
