diff --git a/src/bin/pg_dump/pg_dump_sort.c b/src/bin/pg_dump/pg_dump_sort.c
index fecf80d2..ffaceb93 100644
--- a/src/bin/pg_dump/pg_dump_sort.c
+++ b/src/bin/pg_dump/pg_dump_sort.c
@@ -417,19 +417,6 @@ DOTypeNameCompare(const void *p1, const void *p2)
 				return cmpval;
 		}
 	}
-	else if (obj1->objType == DO_DEFAULT_ACL)
-	{
-		DefaultACLInfo *daclobj1 = *(DefaultACLInfo *const *) p1;
-		DefaultACLInfo *daclobj2 = *(DefaultACLInfo *const *) p2;
-
-		/*
-		 * Sort by defaclrole, per pg_default_acl_role_nsp_obj_index.  The
-		 * (namespace, name) match (defaclnamespace, defaclobjtype).
-		 */
-		cmpval = strcmp(daclobj1->defaclrole, daclobj2->defaclrole);
-		if (cmpval != 0)
-			return cmpval;
-	}
 	else if (obj1->objType == DO_PUBLICATION_REL)
 	{
 		PublicationRelInfo *probj1 = *(PublicationRelInfo *const *) p1;
