diff --git a/src/backend/optimizer/path/costsize.c b/src/backend/optimizer/path/costsize.c
index 5a0ecd80..52ebdd90 100644
--- a/src/backend/optimizer/path/costsize.c
+++ b/src/backend/optimizer/path/costsize.c
@@ -4436,24 +4436,10 @@ cost_subplan(PlannerInfo *root, SubPlan *subplan, Plan *plan)
 {
 	QualCost	sp_cost;
 
-	/*
-	 * Figure any cost for evaluating the testexpr.
-	 *
-	 * Usually, SubPlan nodes are built very early, before we have constructed
-	 * any RelOptInfos for the parent query level, which means the parent root
-	 * does not yet contain enough information to safely consult statistics.
-	 * Therefore, we pass root as NULL here.  cost_qual_eval() is already
-	 * well-equipped to handle a NULL root.
-	 *
-	 * One exception is SubPlan nodes built for the initplans of MIN/MAX
-	 * aggregates from indexes (cf. SS_make_initplan_from_plan).  In this
-	 * case, having a NULL root is safe because testexpr will be NULL.
-	 * Besides, an initplan will by definition not consult anything from the
-	 * parent plan.
-	 */
+	/* Figure any cost for evaluating the testexpr */
 	cost_qual_eval(&sp_cost,
 				   make_ands_implicit((Expr *) subplan->testexpr),
-				   NULL);
+				   root);
 
 	if (subplan->useHashTable)
 	{
