diff --git a/src/backend/utils/adt/jsonb.c b/src/backend/utils/adt/jsonb.c
index b6110aba..e4562b3c 100644
--- a/src/backend/utils/adt/jsonb.c
+++ b/src/backend/utils/adt/jsonb.c
@@ -657,7 +657,7 @@ datum_to_jsonb_internal(Datum val, bool is_null, JsonbInState *result,
 			  tcategory == JSONTYPE_COMPOSITE ||
 			  tcategory == JSONTYPE_JSON ||
 			  tcategory == JSONTYPE_JSONB ||
-			  tcategory == JSONTYPE_CAST))
+			  tcategory == JSONTYPE_JSON))
 	{
 		ereport(ERROR,
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
