diff --git a/src/backend/parser/parse_expr.c b/src/backend/parser/parse_expr.c
index a3356485..b27e5977 100644
--- a/src/backend/parser/parse_expr.c
+++ b/src/backend/parser/parse_expr.c
@@ -4273,9 +4273,6 @@ transformJsonFuncExpr(ParseState *pstate, JsonFuncExpr *func)
 {
 	JsonExpr   *jsexpr;
 	Node	   *path_spec;
-	Oid			pathspec_type;
-	int			pathspec_loc;
-	Node	   *coerced_path_spec;
 	const char *func_name = NULL;
 	JsonFormatType default_format;
 
@@ -4491,21 +4488,17 @@ transformJsonFuncExpr(ParseState *pstate, JsonFuncExpr *func)
 	jsexpr->format = func->context_item->format;
 
 	path_spec = transformExprRecurse(pstate, func->pathspec);
-	pathspec_type = exprType(path_spec);
-	pathspec_loc = exprLocation(path_spec);
-	coerced_path_spec = coerce_to_target_type(pstate, path_spec,
-											  pathspec_type,
-											  JSONPATHOID, -1,
-											  COERCION_EXPLICIT,
-											  COERCE_IMPLICIT_CAST,
-											  pathspec_loc);
-	if (coerced_path_spec == NULL)
+	path_spec = coerce_to_target_type(pstate, path_spec, exprType(path_spec),
+									  JSONPATHOID, -1,
+									  COERCION_EXPLICIT, COERCE_IMPLICIT_CAST,
+									  exprLocation(path_spec));
+	if (path_spec == NULL)
 		ereport(ERROR,
 				(errcode(ERRCODE_DATATYPE_MISMATCH),
 				 errmsg("JSON path expression must be of type %s, not of type %s",
-						"jsonpath", format_type_be(pathspec_type)),
-				 parser_errposition(pstate, pathspec_loc)));
-	jsexpr->path_spec = coerced_path_spec;
+						"jsonpath", format_type_be(exprType(path_spec))),
+				 parser_errposition(pstate, exprLocation(path_spec))));
+	jsexpr->path_spec = path_spec;
 
 	/* Transform and coerce the PASSING arguments to jsonb. */
 	transformJsonPassingArgs(pstate, func_name,
