diff --git a/contrib/spi/refint.c b/contrib/spi/refint.c
index 80b9ef65..2fa96fe0 100644
--- a/contrib/spi/refint.c
+++ b/contrib/spi/refint.c
@@ -12,10 +12,25 @@
 #include "commands/trigger.h"
 #include "executor/spi.h"
 #include "utils/builtins.h"
+#include "utils/memutils.h"
 #include "utils/rel.h"
 
 PG_MODULE_MAGIC;
 
+typedef struct
+{
+	char	   *ident;
+	int			nplans;
+	SPIPlanPtr *splan;
+} EPlan;
+
+static EPlan *FPlans = NULL;
+static int	nFPlans = 0;
+static EPlan *PPlans = NULL;
+static int	nPPlans = 0;
+
+static EPlan *find_plan(char *ident, EPlan **eplan, int *nplans);
+
 /*
  * check_primary_key () -- check that key in tuple being inserted/updated
  *			 references existing tuple in "primary" table.
@@ -41,12 +56,12 @@ check_primary_key(PG_FUNCTION_ARGS)
 	Relation	rel;			/* triggered relation */
 	HeapTuple	tuple = NULL;	/* tuple to return */
 	TupleDesc	tupdesc;		/* tuple description */
-	SPIPlanPtr	pplan;			/* prepared plan */
+	EPlan	   *plan;			/* prepared plan */
 	Oid		   *argtypes = NULL;	/* key types to prepare execution plan */
 	bool		isnull;			/* to know is some column NULL or not */
+	char		ident[2 * NAMEDATALEN]; /* to identify myself */
 	int			ret;
 	int			i;
-	StringInfoData sql;
 
 #ifdef	DEBUG_QUERY
 	elog(DEBUG4, "check_primary_key: Enter Function");
@@ -103,8 +118,16 @@ check_primary_key(PG_FUNCTION_ARGS)
 	 */
 	kvals = (Datum *) palloc(nkeys * sizeof(Datum));
 
-	/* allocate argtypes for preparation */
-	argtypes = (Oid *) palloc(nkeys * sizeof(Oid));
+	/*
+	 * Construct ident string as TriggerName $ TriggeredRelationId and try to
+	 * find prepared execution plan.
+	 */
+	snprintf(ident, sizeof(ident), "%s$%u", trigger->tgname, rel->rd_id);
+	plan = find_plan(ident, &PPlans, &nPPlans);
+
+	/* if there is no plan then allocate argtypes for preparation */
+	if (plan->nplans <= 0)
+		argtypes = (Oid *) palloc(nkeys * sizeof(Oid));
 
 	/* For each column in key ... */
 	for (i = 0; i < nkeys; i++)
@@ -133,36 +156,57 @@ check_primary_key(PG_FUNCTION_ARGS)
 			return PointerGetDatum(tuple);
 		}
 
-		/* Get typeId of column */
-		argtypes[i] = SPI_gettypeid(tupdesc, fnumber);
+		if (plan->nplans <= 0)	/* Get typeId of column */
+			argtypes[i] = SPI_gettypeid(tupdesc, fnumber);
 	}
 
-	initStringInfo(&sql);
-
 	/*
-	 * Construct query: SELECT 1 FROM _referenced_relation_ WHERE Pkey1 = $1
-	 * [AND Pkey2 = $2 [...]]
+	 * If we have to prepare plan ...
 	 */
-	appendStringInfo(&sql, "select 1 from %s where ", relname);
-	for (i = 1; i <= nkeys; i++)
+	if (plan->nplans <= 0)
 	{
-		appendStringInfo(&sql, "%s = $%d ", args[i + nkeys], i);
-		if (i < nkeys)
-			appendStringInfoString(&sql, "and ");
-	}
+		SPIPlanPtr	pplan;
+		StringInfoData sql;
 
-	/* Prepare plan for query */
-	pplan = SPI_prepare(sql.data, nkeys, argtypes);
-	if (pplan == NULL)
-		/* internal error */
-		elog(ERROR, "check_primary_key: SPI_prepare returned %s", SPI_result_code_string(SPI_result));
+		initStringInfo(&sql);
 
-	pfree(sql.data);
+		/*
+		 * Construct query: SELECT 1 FROM _referenced_relation_ WHERE Pkey1 =
+		 * $1 [AND Pkey2 = $2 [...]]
+		 */
+		appendStringInfo(&sql, "select 1 from %s where ", relname);
+		for (i = 1; i <= nkeys; i++)
+		{
+			appendStringInfo(&sql, "%s = $%d ", args[i + nkeys], i);
+			if (i < nkeys)
+				appendStringInfoString(&sql, "and ");
+		}
+
+		/* Prepare plan for query */
+		pplan = SPI_prepare(sql.data, nkeys, argtypes);
+		if (pplan == NULL)
+			/* internal error */
+			elog(ERROR, "check_primary_key: SPI_prepare returned %s", SPI_result_code_string(SPI_result));
+
+		/*
+		 * Remember that SPI_prepare places plan in current memory context -
+		 * so, we have to save plan in TopMemoryContext for later use.
+		 */
+		if (SPI_keepplan(pplan))
+			/* internal error */
+			elog(ERROR, "check_primary_key: SPI_keepplan failed");
+		plan->splan = (SPIPlanPtr *) MemoryContextAlloc(TopMemoryContext,
+														sizeof(SPIPlanPtr));
+		*(plan->splan) = pplan;
+		plan->nplans = 1;
+
+		pfree(sql.data);
+	}
 
 	/*
 	 * Ok, execute prepared plan.
 	 */
-	ret = SPI_execp(pplan, kvals, NULL, 1);
+	ret = SPI_execp(*(plan->splan), kvals, NULL, 1);
 	/* we have no NULLs - so we pass   ^^^^   here */
 
 	if (ret < 0)
@@ -214,15 +258,15 @@ check_foreign_key(PG_FUNCTION_ARGS)
 	HeapTuple	trigtuple = NULL;	/* tuple to being changed */
 	HeapTuple	newtuple = NULL;	/* tuple to return */
 	TupleDesc	tupdesc;		/* tuple description */
-	SPIPlanPtr *splan;			/* prepared plan(s) */
+	EPlan	   *plan;			/* prepared plan(s) */
 	Oid		   *argtypes = NULL;	/* key types to prepare execution plan */
 	bool		isnull;			/* to know is some column NULL or not */
 	bool		isequal = true; /* are keys in both tuples equal (in UPDATE) */
+	char		ident[2 * NAMEDATALEN]; /* to identify myself */
 	int			is_update = 0;
 	int			ret;
 	int			i,
 				r;
-	char	  **args2;
 
 #ifdef DEBUG_QUERY
 	elog(DEBUG4, "check_foreign_key: Enter Function");
@@ -299,8 +343,24 @@ check_foreign_key(PG_FUNCTION_ARGS)
 	 */
 	kvals = (Datum *) palloc(nkeys * sizeof(Datum));
 
-	/* allocate argtypes for preparation */
-	argtypes = (Oid *) palloc(nkeys * sizeof(Oid));
+	/*
+	 * Construct ident string as TriggerName $ TriggeredRelationId and try to
+	 * find prepared execution plan(s).
+	 */
+	snprintf(ident, sizeof(ident), "%s$%u", trigger->tgname, rel->rd_id);
+	plan = find_plan(ident, &FPlans, &nFPlans);
+
+	/* if there is no plan(s) then allocate argtypes for preparation */
+	if (plan->nplans <= 0)
+		argtypes = (Oid *) palloc(nkeys * sizeof(Oid));
+
+	/*
+	 * else - check that we have exactly nrefs plan(s) ready
+	 */
+	else if (plan->nplans != nrefs)
+		/* internal error */
+		elog(ERROR, "%s: check_foreign_key: # of plans changed in meantime",
+			 trigger->tgname);
 
 	/* For each column in key ... */
 	for (i = 0; i < nkeys; i++)
@@ -348,124 +408,141 @@ check_foreign_key(PG_FUNCTION_ARGS)
 				isequal = false;
 		}
 
-		/* Get typeId of column */
-		argtypes[i] = SPI_gettypeid(tupdesc, fnumber);
+		if (plan->nplans <= 0)	/* Get typeId of column */
+			argtypes[i] = SPI_gettypeid(tupdesc, fnumber);
 	}
 	args_temp = args;
 	nargs -= nkeys;
 	args += nkeys;
-	args2 = args;
 
-	splan = (SPIPlanPtr *) palloc(nrefs * sizeof(SPIPlanPtr));
-
-	for (r = 0; r < nrefs; r++)
+	/*
+	 * If we have to prepare plans ...
+	 */
+	if (plan->nplans <= 0)
 	{
-		StringInfoData sql;
 		SPIPlanPtr	pplan;
+		char	  **args2 = args;
 
-		initStringInfo(&sql);
-
-		relname = args2[0];
-
-		/*---------
-		 * For 'R'estrict action we construct SELECT query:
-		 *
-		 *	SELECT 1
-		 *	FROM _referencing_relation_
-		 *	WHERE Fkey1 = $1 [AND Fkey2 = $2 [...]]
-		 *
-		 *	to check is tuple referenced or not.
-		 *---------
-		 */
-		if (action == 'r')
-			appendStringInfo(&sql, "select 1 from %s where ", relname);
-
-		/*---------
-		 * For 'C'ascade action we construct DELETE query
-		 *
-		 *	DELETE
-		 *	FROM _referencing_relation_
-		 *	WHERE Fkey1 = $1 [AND Fkey2 = $2 [...]]
-		 *
-		 * to delete all referencing tuples.
-		 *---------
-		 */
-
-		/*
-		 * Max : Cascade with UPDATE query i create update query that updates
-		 * new key values in referenced tables
-		 */
-
+		plan->splan = (SPIPlanPtr *) MemoryContextAlloc(TopMemoryContext,
+														nrefs * sizeof(SPIPlanPtr));
 
-		else if (action == 'c')
+		for (r = 0; r < nrefs; r++)
 		{
-			if (is_update == 1)
+			StringInfoData sql;
+
+			initStringInfo(&sql);
+
+			relname = args2[0];
+
+			/*---------
+			 * For 'R'estrict action we construct SELECT query:
+			 *
+			 *	SELECT 1
+			 *	FROM _referencing_relation_
+			 *	WHERE Fkey1 = $1 [AND Fkey2 = $2 [...]]
+			 *
+			 *	to check is tuple referenced or not.
+			 *---------
+			 */
+			if (action == 'r')
+				appendStringInfo(&sql, "select 1 from %s where ", relname);
+
+			/*---------
+			 * For 'C'ascade action we construct DELETE query
+			 *
+			 *	DELETE
+			 *	FROM _referencing_relation_
+			 *	WHERE Fkey1 = $1 [AND Fkey2 = $2 [...]]
+			 *
+			 * to delete all referencing tuples.
+			 *---------
+			 */
+
+			/*
+			 * Max : Cascade with UPDATE query i create update query that
+			 * updates new key values in referenced tables
+			 */
+
+
+			else if (action == 'c')
 			{
-				int			fn;
-				char	   *nv;
-				int			k;
+				if (is_update == 1)
+				{
+					int			fn;
+					char	   *nv;
+					int			k;
+
+					appendStringInfo(&sql, "update %s set ", relname);
+					for (k = 1; k <= nkeys; k++)
+					{
+						fn = SPI_fnumber(tupdesc, args_temp[k - 1]);
+						Assert(fn > 0); /* already checked above */
+						nv = SPI_getvalue(newtuple, tupdesc, fn);
+
+						appendStringInfo(&sql, " %s = %s ",
+										 args2[k],
+										 nv ? quote_literal_cstr(nv) : "NULL");
+						if (k < nkeys)
+							appendStringInfoString(&sql, ", ");
+					}
+					appendStringInfoString(&sql, " where ");
+				}
+				else
+					/* DELETE */
+					appendStringInfo(&sql, "delete from %s where ", relname);
+			}
 
+			/*
+			 * For 'S'etnull action we construct UPDATE query - UPDATE
+			 * _referencing_relation_ SET Fkey1 null [, Fkey2 null [...]]
+			 * WHERE Fkey1 = $1 [AND Fkey2 = $2 [...]] - to set key columns in
+			 * all referencing tuples to NULL.
+			 */
+			else if (action == 's')
+			{
 				appendStringInfo(&sql, "update %s set ", relname);
-				for (k = 1; k <= nkeys; k++)
+				for (i = 1; i <= nkeys; i++)
 				{
-					fn = SPI_fnumber(tupdesc, args_temp[k - 1]);
-					Assert(fn > 0); /* already checked above */
-					nv = SPI_getvalue(newtuple, tupdesc, fn);
-
-					appendStringInfo(&sql, " %s = %s ",
-									 args2[k],
-									 nv ? quote_literal_cstr(nv) : "NULL");
-					if (k < nkeys)
+					appendStringInfo(&sql, "%s = null", args2[i]);
+					if (i < nkeys)
 						appendStringInfoString(&sql, ", ");
 				}
 				appendStringInfoString(&sql, " where ");
 			}
-			else
-				/* DELETE */
-				appendStringInfo(&sql, "delete from %s where ", relname);
-		}
 
-		/*
-		 * For 'S'etnull action we construct UPDATE query - UPDATE
-		 * _referencing_relation_ SET Fkey1 null [, Fkey2 null [...]] WHERE
-		 * Fkey1 = $1 [AND Fkey2 = $2 [...]] - to set key columns in all
-		 * referencing tuples to NULL.
-		 */
-		else if (action == 's')
-		{
-			appendStringInfo(&sql, "update %s set ", relname);
+			/* Construct WHERE qual */
 			for (i = 1; i <= nkeys; i++)
 			{
-				appendStringInfo(&sql, "%s = null", args2[i]);
+				appendStringInfo(&sql, "%s = $%d ", args2[i], i);
 				if (i < nkeys)
-					appendStringInfoString(&sql, ", ");
+					appendStringInfoString(&sql, "and ");
 			}
-			appendStringInfoString(&sql, " where ");
-		}
 
-		/* Construct WHERE qual */
-		for (i = 1; i <= nkeys; i++)
-		{
-			appendStringInfo(&sql, "%s = $%d ", args2[i], i);
-			if (i < nkeys)
-				appendStringInfoString(&sql, "and ");
-		}
+			/* Prepare plan for query */
+			pplan = SPI_prepare(sql.data, nkeys, argtypes);
+			if (pplan == NULL)
+				/* internal error */
+				elog(ERROR, "check_foreign_key: SPI_prepare returned %s", SPI_result_code_string(SPI_result));
 
-		/* Prepare plan for query */
-		pplan = SPI_prepare(sql.data, nkeys, argtypes);
-		if (pplan == NULL)
-			/* internal error */
-			elog(ERROR, "check_foreign_key: SPI_prepare returned %s", SPI_result_code_string(SPI_result));
+			/*
+			 * Remember that SPI_prepare places plan in current memory context
+			 * - so, we have to save plan in Top memory context for later use.
+			 */
+			if (SPI_keepplan(pplan))
+				/* internal error */
+				elog(ERROR, "check_foreign_key: SPI_keepplan failed");
 
-		splan[r] = pplan;
+			plan->splan[r] = pplan;
 
-		args2 += nkeys + 1;		/* to the next relation */
+			args2 += nkeys + 1; /* to the next relation */
 
 #ifdef DEBUG_QUERY
-		elog(DEBUG4, "check_foreign_key Debug Query is :  %s ", sql.data);
+			elog(DEBUG4, "check_foreign_key Debug Query is :  %s ", sql.data);
 #endif
 
-		pfree(sql.data);
+			pfree(sql.data);
+		}
+		plan->nplans = nrefs;
 	}
 
 	/*
@@ -490,7 +567,9 @@ check_foreign_key(PG_FUNCTION_ARGS)
 
 		relname = args[0];
 
-		ret = SPI_execp(splan[r], kvals, NULL, tcount);
+		snprintf(ident, sizeof(ident), "%s$%u", trigger->tgname, rel->rd_id);
+		plan = find_plan(ident, &FPlans, &nFPlans);
+		ret = SPI_execp(plan->splan[r], kvals, NULL, tcount);
 		/* we have no NULLs - so we pass   ^^^^  here */
 
 		if (ret < 0)
@@ -523,3 +602,46 @@ check_foreign_key(PG_FUNCTION_ARGS)
 
 	return PointerGetDatum((newtuple == NULL) ? trigtuple : newtuple);
 }
+
+static EPlan *
+find_plan(char *ident, EPlan **eplan, int *nplans)
+{
+	EPlan	   *newp;
+	int			i;
+	MemoryContext oldcontext;
+
+	/*
+	 * All allocations done for the plans need to happen in a session-safe
+	 * context.
+	 */
+	oldcontext = MemoryContextSwitchTo(TopMemoryContext);
+
+	if (*nplans > 0)
+	{
+		for (i = 0; i < *nplans; i++)
+		{
+			if (strcmp((*eplan)[i].ident, ident) == 0)
+				break;
+		}
+		if (i != *nplans)
+		{
+			MemoryContextSwitchTo(oldcontext);
+			return (*eplan + i);
+		}
+		*eplan = (EPlan *) repalloc(*eplan, (i + 1) * sizeof(EPlan));
+		newp = *eplan + i;
+	}
+	else
+	{
+		newp = *eplan = (EPlan *) palloc(sizeof(EPlan));
+		(*nplans) = i = 0;
+	}
+
+	newp->ident = pstrdup(ident);
+	newp->nplans = 0;
+	newp->splan = NULL;
+	(*nplans)++;
+
+	MemoryContextSwitchTo(oldcontext);
+	return newp;
+}
