diff --git a/src/backend/tsearch/wparser_def.c b/src/backend/tsearch/wparser_def.c
index 99e4ee73..2b801564 100644
--- a/src/backend/tsearch/wparser_def.c
+++ b/src/backend/tsearch/wparser_def.c
@@ -2628,9 +2628,6 @@ prsd_headline(PG_FUNCTION_ARGS)
 	int			max_fragments = 0;
 	bool		highlightall = false;
 	ListCell   *l;
-	size_t		startsellen;
-	size_t		stopsellen;
-	size_t		fragdelimlen;
 
 	/* Extract configuration option values */
 	prs->startsel = NULL;
@@ -2720,24 +2717,9 @@ prsd_headline(PG_FUNCTION_ARGS)
 		prs->fragdelim = pstrdup(" ... ");
 
 	/* Caller will need these lengths, too */
-	startsellen = strlen(prs->startsel);
-	stopsellen = strlen(prs->stopsel);
-	fragdelimlen = strlen(prs->fragdelim);
-	if (startsellen > PG_INT16_MAX)
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("value for \"%s\" is too long", "StartSel")));
-	if (stopsellen > PG_INT16_MAX)
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("value for \"%s\" is too long", "StopSel")));
-	if (fragdelimlen > PG_INT16_MAX)
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("value for \"%s\" is too long", "FragmentDelimiter")));
-	prs->startsellen = startsellen;
-	prs->stopsellen = stopsellen;
-	prs->fragdelimlen = fragdelimlen;
+	prs->startsellen = strlen(prs->startsel);
+	prs->stopsellen = strlen(prs->stopsel);
+	prs->fragdelimlen = strlen(prs->fragdelim);
 
 	PG_RETURN_POINTER(prs);
 }
