diff --git a/src/backend/executor/nodeWindowAgg.c b/src/backend/executor/nodeWindowAgg.c
index eb278a25..96c117c5 100644
--- a/src/backend/executor/nodeWindowAgg.c
+++ b/src/backend/executor/nodeWindowAgg.c
@@ -37,7 +37,6 @@
 #include "catalog/objectaccess.h"
 #include "catalog/pg_aggregate.h"
 #include "catalog/pg_proc.h"
-#include "common/int.h"
 #include "executor/executor.h"
 #include "executor/nodeWindowAgg.h"
 #include "miscadmin.h"
@@ -1425,21 +1424,12 @@ row_is_in_frame(WindowAggState *winstate, int64 pos, TupleTableSlot *slot)
 		if (frameOptions & FRAMEOPTION_ROWS)
 		{
 			int64		offset = DatumGetInt64(winstate->endOffsetValue);
-			int64		frameendpos = 0;
 
 			/* rows after current row + offset are out of frame */
 			if (frameOptions & FRAMEOPTION_END_OFFSET_PRECEDING)
 				offset = -offset;
 
-			/*
-			 * If we have an overflow, it means the frame end is beyond the
-			 * range of int64.  Since currentpos >= 0, this can only be a
-			 * positive overflow.  We treat this as meaning that the frame
-			 * extends to end of partition.
-			 */
-			if (!pg_add_s64_overflow(winstate->currentpos, offset,
-									 &frameendpos) &&
-				pos > frameendpos)
+			if (pos > winstate->currentpos + offset)
 				return -1;
 		}
 		else if (frameOptions & (FRAMEOPTION_RANGE | FRAMEOPTION_GROUPS))
@@ -1574,16 +1564,7 @@ update_frameheadpos(WindowAggState *winstate)
 			if (frameOptions & FRAMEOPTION_START_OFFSET_PRECEDING)
 				offset = -offset;
 
-			/*
-			 * If we have an overflow, it means the frame head is beyond the
-			 * range of int64.  Since currentpos >= 0, this can only be a
-			 * positive overflow.  We treat this as being beyond end of
-			 * partition.
-			 */
-			if (pg_add_s64_overflow(winstate->currentpos, offset,
-									&winstate->frameheadpos))
-				winstate->frameheadpos = PG_INT64_MAX;
-
+			winstate->frameheadpos = winstate->currentpos + offset;
 			/* frame head can't go before first row */
 			if (winstate->frameheadpos < 0)
 				winstate->frameheadpos = 0;
@@ -1695,21 +1676,12 @@ update_frameheadpos(WindowAggState *winstate)
 			 * framehead_slot empty.
 			 */
 			int64		offset = DatumGetInt64(winstate->startOffsetValue);
-			int64		minheadgroup = 0;
+			int64		minheadgroup;
 
 			if (frameOptions & FRAMEOPTION_START_OFFSET_PRECEDING)
 				minheadgroup = winstate->currentgroup - offset;
 			else
-			{
-				/*
-				 * If we have an overflow, it means the target group is beyond
-				 * the range of int64.  We treat this as "infinity", which
-				 * ensures the loop below advances to end of partition.
-				 */
-				if (pg_add_s64_overflow(winstate->currentgroup, offset,
-										&minheadgroup))
-					minheadgroup = PG_INT64_MAX;
-			}
+				minheadgroup = winstate->currentgroup + offset;
 
 			tuplestore_select_read_pointer(winstate->buffer,
 										   winstate->framehead_ptr);
@@ -1846,18 +1818,7 @@ update_frametailpos(WindowAggState *winstate)
 			if (frameOptions & FRAMEOPTION_END_OFFSET_PRECEDING)
 				offset = -offset;
 
-			/*
-			 * If we have an overflow, it means the frame tail is beyond the
-			 * range of int64.  Since currentpos >= 0, this can only be a
-			 * positive overflow.  We treat this as being beyond end of
-			 * partition.
-			 */
-			if (pg_add_s64_overflow(winstate->currentpos, offset,
-									&winstate->frametailpos) ||
-				pg_add_s64_overflow(winstate->frametailpos, 1,
-									&winstate->frametailpos))
-				winstate->frametailpos = PG_INT64_MAX;
-
+			winstate->frametailpos = winstate->currentpos + offset + 1;
 			/* smallest allowable value of frametailpos is 0 */
 			if (winstate->frametailpos < 0)
 				winstate->frametailpos = 0;
@@ -1969,21 +1930,12 @@ update_frametailpos(WindowAggState *winstate)
 			 * leave frametailpos = end+1 and frametail_slot empty.
 			 */
 			int64		offset = DatumGetInt64(winstate->endOffsetValue);
-			int64		maxtailgroup = 0;
+			int64		maxtailgroup;
 
 			if (frameOptions & FRAMEOPTION_END_OFFSET_PRECEDING)
 				maxtailgroup = winstate->currentgroup - offset;
 			else
-			{
-				/*
-				 * If we have an overflow, it means the target group is beyond
-				 * the range of int64.  We treat this as "infinity", which
-				 * ensures the loop below advances to end of partition.
-				 */
-				if (pg_add_s64_overflow(winstate->currentgroup, offset,
-										&maxtailgroup))
-					maxtailgroup = PG_INT64_MAX;
-			}
+				maxtailgroup = winstate->currentgroup + offset;
 
 			tuplestore_select_read_pointer(winstate->buffer,
 										   winstate->frametail_ptr);
