diff --git a/src/backend/parser/parse_agg.c b/src/backend/parser/parse_agg.c
index cf449efd..be88909d 100644
--- a/src/backend/parser/parse_agg.c
+++ b/src/backend/parser/parse_agg.c
@@ -786,32 +786,6 @@ check_agg_arguments_walker(Node *node,
 					 parser_errposition(context->pstate,
 										((WindowFunc *) node)->location)));
 	}
-
-	if (IsA(node, RangeTblEntry))
-	{
-		/*
-		 * CTE references act similarly to Vars of the CTE's level.  Without
-		 * this we might conclude that the Agg can be evaluated above the CTE,
-		 * leading to trouble.
-		 */
-		RangeTblEntry *rte = (RangeTblEntry *) node;
-
-		if (rte->rtekind == RTE_CTE)
-		{
-			int			ctelevelsup = rte->ctelevelsup;
-
-			/* convert levelsup to frame of reference of original query */
-			ctelevelsup -= context->sublevels_up;
-			/* ignore local CTEs of subqueries */
-			if (ctelevelsup >= 0)
-			{
-				if (context->min_varlevel < 0 ||
-					context->min_varlevel > ctelevelsup)
-					context->min_varlevel = ctelevelsup;
-			}
-		}
-		return false;			/* allow range_table_walker to continue */
-	}
 	if (IsA(node, Query))
 	{
 		/* Recurse into subselects */
@@ -821,7 +795,7 @@ check_agg_arguments_walker(Node *node,
 		result = query_tree_walker((Query *) node,
 								   check_agg_arguments_walker,
 								   (void *) context,
-								   QTW_EXAMINE_RTES_BEFORE);
+								   0);
 		context->sublevels_up--;
 		return result;
 	}
