diff --git a/src/backend/parser/parse_agg.c b/src/backend/parser/parse_agg.c
index f7c004dc..cf449efd 100644
--- a/src/backend/parser/parse_agg.c
+++ b/src/backend/parser/parse_agg.c
@@ -37,8 +37,6 @@ typedef struct
 	ParseState *pstate;
 	int			min_varlevel;
 	int			min_agglevel;
-	int			min_ctelevel;
-	RangeTblEntry *min_cte;
 	int			sublevels_up;
 } check_agg_arguments_context;
 
@@ -58,8 +56,7 @@ typedef struct
 static int	check_agg_arguments(ParseState *pstate,
 								List *directargs,
 								List *args,
-								Expr *filter,
-								int agglocation);
+								Expr *filter);
 static bool check_agg_arguments_walker(Node *node,
 									   check_agg_arguments_context *context);
 static void check_ungrouped_columns(Node *node, ParseState *pstate, Query *qry,
@@ -337,8 +334,7 @@ check_agglevels_and_constraints(ParseState *pstate, Node *expr)
 	min_varlevel = check_agg_arguments(pstate,
 									   directargs,
 									   args,
-									   filter,
-									   location);
+									   filter);
 
 	*p_levelsup = min_varlevel;
 
@@ -640,8 +636,7 @@ static int
 check_agg_arguments(ParseState *pstate,
 					List *directargs,
 					List *args,
-					Expr *filter,
-					int agglocation)
+					Expr *filter)
 {
 	int			agglevel;
 	check_agg_arguments_context context;
@@ -649,8 +644,6 @@ check_agg_arguments(ParseState *pstate,
 	context.pstate = pstate;
 	context.min_varlevel = -1;	/* signifies nothing found yet */
 	context.min_agglevel = -1;
-	context.min_ctelevel = -1;
-	context.min_cte = NULL;
 	context.sublevels_up = 0;
 
 	(void) check_agg_arguments_walker((Node *) args, &context);
@@ -688,20 +681,6 @@ check_agg_arguments(ParseState *pstate,
 				 parser_errposition(pstate, aggloc)));
 	}
 
-	/*
-	 * If there's a non-local CTE that's below the aggregate's semantic level,
-	 * complain.  It's not quite clear what we should do to fix up such a case
-	 * (treating the CTE reference like a Var seems wrong), and it's also
-	 * unclear whether there is a real-world use for such cases.
-	 */
-	if (context.min_ctelevel >= 0 && context.min_ctelevel < agglevel)
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("outer-level aggregate cannot use a nested CTE"),
-				 errdetail("CTE \"%s\" is below the aggregate's semantic level.",
-						   context.min_cte->eref->aliasname),
-				 parser_errposition(pstate, agglocation)));
-
 	/*
 	 * Now check for vars/aggs in the direct arguments, and throw error if
 	 * needed.  Note that we allow a Var of the agg's semantic level, but not
@@ -715,7 +694,6 @@ check_agg_arguments(ParseState *pstate,
 	{
 		context.min_varlevel = -1;
 		context.min_agglevel = -1;
-		context.min_ctelevel = -1;
 		(void) check_agg_arguments_walker((Node *) directargs, &context);
 		if (context.min_varlevel >= 0 && context.min_varlevel < agglevel)
 			ereport(ERROR,
@@ -731,13 +709,6 @@ check_agg_arguments(ParseState *pstate,
 					 parser_errposition(pstate,
 										locate_agg_of_level((Node *) directargs,
 															context.min_agglevel))));
-		if (context.min_ctelevel >= 0 && context.min_ctelevel < agglevel)
-			ereport(ERROR,
-					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					 errmsg("outer-level aggregate cannot use a nested CTE"),
-					 errdetail("CTE \"%s\" is below the aggregate's semantic level.",
-							   context.min_cte->eref->aliasname),
-					 parser_errposition(pstate, agglocation)));
 	}
 	return agglevel;
 }
@@ -818,6 +789,11 @@ check_agg_arguments_walker(Node *node,
 
 	if (IsA(node, RangeTblEntry))
 	{
+		/*
+		 * CTE references act similarly to Vars of the CTE's level.  Without
+		 * this we might conclude that the Agg can be evaluated above the CTE,
+		 * leading to trouble.
+		 */
 		RangeTblEntry *rte = (RangeTblEntry *) node;
 
 		if (rte->rtekind == RTE_CTE)
@@ -829,12 +805,9 @@ check_agg_arguments_walker(Node *node,
 			/* ignore local CTEs of subqueries */
 			if (ctelevelsup >= 0)
 			{
-				if (context->min_ctelevel < 0 ||
-					context->min_ctelevel > ctelevelsup)
-				{
-					context->min_ctelevel = ctelevelsup;
-					context->min_cte = rte;
-				}
+				if (context->min_varlevel < 0 ||
+					context->min_varlevel > ctelevelsup)
+					context->min_varlevel = ctelevelsup;
 			}
 		}
 		return false;			/* allow range_table_walker to continue */
