diff --git a/src/backend/rewrite/rewriteHandler.c b/src/backend/rewrite/rewriteHandler.c
index ed80cb2e..beca0a9c 100644
--- a/src/backend/rewrite/rewriteHandler.c
+++ b/src/backend/rewrite/rewriteHandler.c
@@ -590,10 +590,7 @@ rewriteRuleAction(Query *parsetree,
 			}
 		}
 
-		/*
-		 * OK, it's safe to combine the CTE lists.  Beware that RewriteQuery
-		 * knows we concatenate the lists in this order.
-		 */
+		/* OK, it's safe to combine the CTE lists */
 		sub_action->cteList = list_concat(sub_action->cteList,
 										  copyObject(parsetree->cteList));
 		/* ... and don't forget about the associated flags */
@@ -3863,13 +3860,9 @@ rewriteTargetView(Query *parsetree, Relation view)
  * orig_rt_length is the length of the originating query's rtable, for product
  * queries created by fireRules(), and 0 otherwise.  This is used to skip any
  * already-processed VALUES RTEs from the original query.
- *
- * num_ctes_processed is the number of CTEs at the end of the query's cteList
- * that have already been rewritten, and must not be rewritten again.
  */
 static List *
-RewriteQuery(Query *parsetree, List *rewrite_events, int orig_rt_length,
-			 int num_ctes_processed)
+RewriteQuery(Query *parsetree, List *rewrite_events, int orig_rt_length)
 {
 	CmdType		event = parsetree->commandType;
 	bool		instead = false;
@@ -3883,29 +3876,17 @@ RewriteQuery(Query *parsetree, List *rewrite_events, int orig_rt_length,
 	 * First, recursively process any insert/update/delete/merge statements in
 	 * WITH clauses.  (We have to do this first because the WITH clauses may
 	 * get copied into rule actions below.)
-	 *
-	 * Any new WITH clauses from rule actions are processed when we recurse
-	 * into product queries below.  However, when recursing, we must take care
-	 * to avoid rewriting a CTE query more than once (because expanding
-	 * generated columns in the targetlist more than once would fail).  Since
-	 * new CTEs from product queries are added to the start of the list (see
-	 * rewriteRuleAction), we just skip the last num_ctes_processed items.
 	 */
 	foreach(lc1, parsetree->cteList)
 	{
 		CommonTableExpr *cte = lfirst_node(CommonTableExpr, lc1);
 		Query	   *ctequery = castNode(Query, cte->ctequery);
-		int			i = foreach_current_index(lc1);
 		List	   *newstuff;
 
-		/* Skip already-processed CTEs at the end of the list */
-		if (i >= list_length(parsetree->cteList) - num_ctes_processed)
-			break;
-
 		if (ctequery->commandType == CMD_SELECT)
 			continue;
 
-		newstuff = RewriteQuery(ctequery, rewrite_events, 0, 0);
+		newstuff = RewriteQuery(ctequery, rewrite_events, 0);
 
 		/*
 		 * Currently we can only handle unconditional, single-statement DO
@@ -3965,7 +3946,6 @@ RewriteQuery(Query *parsetree, List *rewrite_events, int orig_rt_length,
 					 errmsg("multi-statement DO INSTEAD rules are not supported for data-modifying statements in WITH")));
 		}
 	}
-	num_ctes_processed = list_length(parsetree->cteList);
 
 	/*
 	 * If the statement is an insert, update, delete, or merge, adjust its
@@ -4297,8 +4277,7 @@ RewriteQuery(Query *parsetree, List *rewrite_events, int orig_rt_length,
 				newstuff = RewriteQuery(pt, rewrite_events,
 										pt == parsetree ?
 										orig_rt_length :
-										product_orig_rt_length,
-										num_ctes_processed);
+										product_orig_rt_length);
 				rewritten = list_concat(rewritten, newstuff);
 			}
 
@@ -4450,7 +4429,7 @@ QueryRewrite(Query *parsetree)
 	 *
 	 * Apply all non-SELECT rules possibly getting 0 or many queries
 	 */
-	querylist = RewriteQuery(parsetree, NIL, 0, 0);
+	querylist = RewriteQuery(parsetree, NIL, 0);
 
 	/*
 	 * Step 2
