diff --git a/src/backend/executor/nodeWindowAgg.c b/src/backend/executor/nodeWindowAgg.c
index 96c117c54..3221fa152 100644
--- a/src/backend/executor/nodeWindowAgg.c
+++ b/src/backend/executor/nodeWindowAgg.c
@@ -2296,23 +2296,6 @@ ExecWindowAgg(PlanState *pstate)
 				 */
 				if (winstate->use_pass_through)
 				{
-					/*
-					 * When switching into a pass-through mode, we'd better
-					 * NULLify the aggregate results as these are no longer
-					 * updated and NULLifying them avoids the old stale
-					 * results lingering.  Some of these might be byref types
-					 * so we can't have them pointing to free'd memory.  The
-					 * planner insisted that quals used in the runcondition
-					 * are strict, so the top-level WindowAgg will always
-					 * filter these NULLs out in the filter clause.
-					 */
-					numfuncs = winstate->numfuncs;
-					for (i = 0; i < numfuncs; i++)
-					{
-						econtext->ecxt_aggvalues[i] = (Datum) 0;
-						econtext->ecxt_aggnulls[i] = true;
-					}
-
 					/*
 					 * STRICT pass-through mode is required for the top window
 					 * when there is a PARTITION BY clause.  Otherwise we must
@@ -2327,6 +2310,24 @@ ExecWindowAgg(PlanState *pstate)
 					else
 					{
 						winstate->status = WINDOWAGG_PASSTHROUGH;
+
+						/*
+						 * If we're not the top-window, we'd better NULLify
+						 * the aggregate results.  In pass-through mode we no
+						 * longer update these and this avoids the old stale
+						 * results lingering.  Some of these might be byref
+						 * types so we can't have them pointing to free'd
+						 * memory.  The planner insisted that quals used in
+						 * the runcondition are strict, so the top-level
+						 * WindowAgg will filter these NULLs out in the filter
+						 * clause.
+						 */
+						numfuncs = winstate->numfuncs;
+						for (i = 0; i < numfuncs; i++)
+						{
+							econtext->ecxt_aggvalues[i] = (Datum) 0;
+							econtext->ecxt_aggnulls[i] = true;
+						}
 					}
 				}
 				else
