diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..9161682 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -777,10 +777,8 @@ date_ne_timestamp(PG_FUNCTION_ARGS)
 Datum
 date_lt_timestamp(PG_FUNCTION_ARGS)
 {
-	DateADT		dateVal = PG_GETARG_DATEADT(0);
-	Timestamp	dt2 = PG_GETARG_TIMESTAMP(1);
-
-	PG_RETURN_BOOL(date_cmp_timestamp_internal(dateVal, dt2) < 0);
+	elog(ERROR, "date_lt_timestamp is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
