diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..f48992c 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -202,26 +202,11 @@ date_out(PG_FUNCTION_ARGS)
 	PG_RETURN_CSTRING(result);
 }
 
-/*
- *		date_recv			- converts external binary format to date
- */
 Datum
 date_recv(PG_FUNCTION_ARGS)
 {
-	StringInfo	buf = (StringInfo) PG_GETARG_POINTER(0);
-	DateADT		result;
-
-	result = (DateADT) pq_getmsgint(buf, sizeof(DateADT));
-
-	/* Limit to the same range that date_in() accepts. */
-	if (DATE_NOT_FINITE(result))
-		 /* ok */ ;
-	else if (!IS_VALID_DATE(result))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("date out of range")));
-
-	PG_RETURN_DATEADT(result);
+	elog(ERROR, "date_recv is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
