diff --git a/src/backend/utils/adt/timestamp.c b/src/backend/utils/adt/timestamp.c
index b57739b..a991b05 100644
--- a/src/backend/utils/adt/timestamp.c
+++ b/src/backend/utils/adt/timestamp.c
@@ -4552,75 +4552,11 @@ timestamptz_age(PG_FUNCTION_ARGS)
  *---------------------------------------------------------*/
 
 
-/* timestamp_bin()
- * Bin timestamp into specified interval.
- */
 Datum
 timestamp_bin(PG_FUNCTION_ARGS)
 {
-	Interval   *stride = PG_GETARG_INTERVAL_P(0);
-	Timestamp	timestamp = PG_GETARG_TIMESTAMP(1);
-	Timestamp	origin = PG_GETARG_TIMESTAMP(2);
-	Timestamp	result,
-				stride_usecs,
-				tm_diff,
-				tm_modulo,
-				tm_delta;
-
-	if (TIMESTAMP_NOT_FINITE(timestamp))
-		PG_RETURN_TIMESTAMP(timestamp);
-
-	if (TIMESTAMP_NOT_FINITE(origin))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("origin out of range")));
-
-	if (INTERVAL_NOT_FINITE(stride))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("timestamps cannot be binned into infinite intervals")));
-
-	if (stride->month != 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("timestamps cannot be binned into intervals containing months or years")));
-
-	if (unlikely(pg_mul_s64_overflow(stride->day, USECS_PER_DAY, &stride_usecs)) ||
-		unlikely(pg_add_s64_overflow(stride_usecs, stride->time, &stride_usecs)))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("interval out of range")));
-
-	if (stride_usecs <= 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("stride must be greater than zero")));
-
-	if (unlikely(pg_sub_s64_overflow(timestamp, origin, &tm_diff)))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("interval out of range")));
-
-	/* These calculations cannot overflow */
-	tm_modulo = tm_diff % stride_usecs;
-	tm_delta = tm_diff - tm_modulo;
-	result = origin + tm_delta;
-
-	/*
-	 * We want to round towards -infinity, not 0, when tm_diff is negative and
-	 * not a multiple of stride_usecs.  This adjustment *can* cause overflow,
-	 * since the result might now be out of the range origin .. timestamp.
-	 */
-	if (tm_modulo < 0)
-	{
-		if (unlikely(pg_sub_s64_overflow(result, stride_usecs, &result)) ||
-			!IS_VALID_TIMESTAMP(result))
-			ereport(ERROR,
-					(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-					 errmsg("timestamp out of range")));
-	}
-
-	PG_RETURN_TIMESTAMP(result);
+	elog(ERROR, "timestamp_bin is not implemented");
+	PG_RETURN_NULL();
 }
 
 /* timestamp_trunc()
@@ -4757,75 +4693,11 @@ timestamp_trunc(PG_FUNCTION_ARGS)
 	PG_RETURN_TIMESTAMP(result);
 }
 
-/* timestamptz_bin()
- * Bin timestamptz into specified interval using specified origin.
- */
 Datum
 timestamptz_bin(PG_FUNCTION_ARGS)
 {
-	Interval   *stride = PG_GETARG_INTERVAL_P(0);
-	TimestampTz timestamp = PG_GETARG_TIMESTAMPTZ(1);
-	TimestampTz origin = PG_GETARG_TIMESTAMPTZ(2);
-	TimestampTz result,
-				stride_usecs,
-				tm_diff,
-				tm_modulo,
-				tm_delta;
-
-	if (TIMESTAMP_NOT_FINITE(timestamp))
-		PG_RETURN_TIMESTAMPTZ(timestamp);
-
-	if (TIMESTAMP_NOT_FINITE(origin))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("origin out of range")));
-
-	if (INTERVAL_NOT_FINITE(stride))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("timestamps cannot be binned into infinite intervals")));
-
-	if (stride->month != 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("timestamps cannot be binned into intervals containing months or years")));
-
-	if (unlikely(pg_mul_s64_overflow(stride->day, USECS_PER_DAY, &stride_usecs)) ||
-		unlikely(pg_add_s64_overflow(stride_usecs, stride->time, &stride_usecs)))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("interval out of range")));
-
-	if (stride_usecs <= 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("stride must be greater than zero")));
-
-	if (unlikely(pg_sub_s64_overflow(timestamp, origin, &tm_diff)))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("interval out of range")));
-
-	/* These calculations cannot overflow */
-	tm_modulo = tm_diff % stride_usecs;
-	tm_delta = tm_diff - tm_modulo;
-	result = origin + tm_delta;
-
-	/*
-	 * We want to round towards -infinity, not 0, when tm_diff is negative and
-	 * not a multiple of stride_usecs.  This adjustment *can* cause overflow,
-	 * since the result might now be out of the range origin .. timestamp.
-	 */
-	if (tm_modulo < 0)
-	{
-		if (unlikely(pg_sub_s64_overflow(result, stride_usecs, &result)) ||
-			!IS_VALID_TIMESTAMP(result))
-			ereport(ERROR,
-					(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-					 errmsg("timestamp out of range")));
-	}
-
-	PG_RETURN_TIMESTAMPTZ(result);
+	elog(ERROR, "timestamptz_bin is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
