diff --git a/src/backend/utils/adt/network.c b/src/backend/utils/adt/network.c
index 640fc37..c56bb3f 100644
--- a/src/backend/utils/adt/network.c
+++ b/src/backend/utils/adt/network.c
@@ -1265,20 +1265,8 @@ network_masklen(PG_FUNCTION_ARGS)
 Datum
 network_family(PG_FUNCTION_ARGS)
 {
-	inet	   *ip = PG_GETARG_INET_PP(0);
-
-	switch (ip_family(ip))
-	{
-		case PGSQL_AF_INET:
-			PG_RETURN_INT32(4);
-			break;
-		case PGSQL_AF_INET6:
-			PG_RETURN_INT32(6);
-			break;
-		default:
-			PG_RETURN_INT32(0);
-			break;
-	}
+	elog(ERROR, "network_family is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
