diff --git a/src/backend/utils/adt/float.c b/src/backend/utils/adt/float.c
index 22421cb..3d4a70b 100644
--- a/src/backend/utils/adt/float.c
+++ b/src/backend/utils/adt/float.c
@@ -3643,30 +3643,8 @@ float8_covar_samp(PG_FUNCTION_ARGS)
 Datum
 float8_corr(PG_FUNCTION_ARGS)
 {
-	ArrayType  *transarray = PG_GETARG_ARRAYTYPE_P(0);
-	float8	   *transvalues;
-	float8		N,
-				Sxx,
-				Syy,
-				Sxy;
-
-	transvalues = check_float8_array(transarray, "float8_corr", 6);
-	N = transvalues[0];
-	Sxx = transvalues[2];
-	Syy = transvalues[4];
-	Sxy = transvalues[5];
-
-	/* if N is 0 we should return NULL */
-	if (N < 1.0)
-		PG_RETURN_NULL();
-
-	/* Note that Sxx and Syy are guaranteed to be non-negative */
-
-	/* per spec, return NULL for horizontal and vertical lines */
-	if (Sxx == 0 || Syy == 0)
-		PG_RETURN_NULL();
-
-	PG_RETURN_FLOAT8(Sxy / sqrt(Sxx * Syy));
+	elog(ERROR, "float8_corr is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
