diff --git a/src/backend/utils/adt/arrayfuncs.c b/src/backend/utils/adt/arrayfuncs.c
index 75cc4ee..530fbf3 100644
--- a/src/backend/utils/adt/arrayfuncs.c
+++ b/src/backend/utils/adt/arrayfuncs.c
@@ -5910,79 +5910,18 @@ typedef struct generate_subscripts_fctx
 	bool		reverse;
 } generate_subscripts_fctx;
 
-/*
- * generate_subscripts(array anyarray, dim int [, reverse bool])
- *		Returns all subscripts of the array for any dimension
- */
 Datum
 generate_subscripts(PG_FUNCTION_ARGS)
 {
-	FuncCallContext *funcctx;
-	MemoryContext oldcontext;
-	generate_subscripts_fctx *fctx;
-
-	/* stuff done only on the first call of the function */
-	if (SRF_IS_FIRSTCALL())
-	{
-		AnyArrayType *v = PG_GETARG_ANY_ARRAY_P(0);
-		int			reqdim = PG_GETARG_INT32(1);
-		int		   *lb,
-				   *dimv;
-
-		/* create a function context for cross-call persistence */
-		funcctx = SRF_FIRSTCALL_INIT();
-
-		/* Sanity check: does it look like an array at all? */
-		if (AARR_NDIM(v) <= 0 || AARR_NDIM(v) > MAXDIM)
-			SRF_RETURN_DONE(funcctx);
-
-		/* Sanity check: was the requested dim valid */
-		if (reqdim <= 0 || reqdim > AARR_NDIM(v))
-			SRF_RETURN_DONE(funcctx);
-
-		/*
-		 * switch to memory context appropriate for multiple function calls
-		 */
-		oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
-		fctx = (generate_subscripts_fctx *) palloc(sizeof(generate_subscripts_fctx));
-
-		lb = AARR_LBOUND(v);
-		dimv = AARR_DIMS(v);
-
-		fctx->lower = lb[reqdim - 1];
-		fctx->upper = dimv[reqdim - 1] + lb[reqdim - 1] - 1;
-		fctx->reverse = (PG_NARGS() < 3) ? false : PG_GETARG_BOOL(2);
-
-		funcctx->user_fctx = fctx;
-
-		MemoryContextSwitchTo(oldcontext);
-	}
-
-	funcctx = SRF_PERCALL_SETUP();
-
-	fctx = funcctx->user_fctx;
-
-	if (fctx->lower <= fctx->upper)
-	{
-		if (!fctx->reverse)
-			SRF_RETURN_NEXT(funcctx, Int32GetDatum(fctx->lower++));
-		else
-			SRF_RETURN_NEXT(funcctx, Int32GetDatum(fctx->upper--));
-	}
-	else
-		/* done when there are no more elements left */
-		SRF_RETURN_DONE(funcctx);
+	elog(ERROR, "generate_subscripts is not implemented");
+	PG_RETURN_NULL();
 }
 
-/*
- * generate_subscripts_nodir
- *		Implements the 2-argument version of generate_subscripts
- */
 Datum
 generate_subscripts_nodir(PG_FUNCTION_ARGS)
 {
-	/* just call the other one -- it can handle both cases */
-	return generate_subscripts(fcinfo);
+	elog(ERROR, "generate_subscripts_nodir is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
