diff --git a/src/backend/utils/adt/oracle_compat.c b/src/backend/utils/adt/oracle_compat.c
index f93077a..c162ef9 100644
--- a/src/backend/utils/adt/oracle_compat.c
+++ b/src/backend/utils/adt/oracle_compat.c
@@ -93,37 +93,11 @@ upper(PG_FUNCTION_ARGS)
 }
 
 
-/********************************************************************
- *
- * initcap
- *
- * Syntax:
- *
- *	 text initcap(text string)
- *
- * Purpose:
- *
- *	 Returns string, with first letter of each word in uppercase, all
- *	 other letters in lowercase. A word is defined as a sequence of
- *	 alphanumeric characters, delimited by non-alphanumeric
- *	 characters.
- *
- ********************************************************************/
-
 Datum
 initcap(PG_FUNCTION_ARGS)
 {
-	text	   *in_string = PG_GETARG_TEXT_PP(0);
-	char	   *out_string;
-	text	   *result;
-
-	out_string = str_initcap(VARDATA_ANY(in_string),
-							 VARSIZE_ANY_EXHDR(in_string),
-							 PG_GET_COLLATION());
-	result = cstring_to_text(out_string);
-	pfree(out_string);
-
-	PG_RETURN_TEXT_P(result);
+	elog(ERROR, "initcap is not implemented");
+	PG_RETURN_NULL();
 }
 
 
