diff --git a/src/backend/utils/adt/jsonfuncs.c b/src/backend/utils/adt/jsonfuncs.c
index 4605234..77a5314 100644
--- a/src/backend/utils/adt/jsonfuncs.c
+++ b/src/backend/utils/adt/jsonfuncs.c
@@ -550,86 +550,11 @@ makeJsonLexContext(JsonLexContext *lex, text *json, bool need_escapes)
 										need_escapes);
 }
 
-/*
- * SQL function json_object_keys
- *
- * Returns the set of keys for the object argument.
- *
- * This SRF operates in value-per-call mode. It processes the
- * object during the first call, and the keys are simply stashed
- * in an array, whose size is expanded as necessary. This is probably
- * safe enough for a list of keys of a single object, since they are
- * limited in size to NAMEDATALEN and the number of keys is unlikely to
- * be so huge that it has major memory implications.
- */
 Datum
 jsonb_object_keys(PG_FUNCTION_ARGS)
 {
-	FuncCallContext *funcctx;
-	OkeysState *state;
-
-	if (SRF_IS_FIRSTCALL())
-	{
-		MemoryContext oldcontext;
-		Jsonb	   *jb = PG_GETARG_JSONB_P(0);
-		bool		skipNested = false;
-		JsonbIterator *it;
-		JsonbValue	v;
-		JsonbIteratorToken r;
-
-		if (JB_ROOT_IS_SCALAR(jb))
-			ereport(ERROR,
-					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-					 errmsg("cannot call %s on a scalar",
-							"jsonb_object_keys")));
-		else if (JB_ROOT_IS_ARRAY(jb))
-			ereport(ERROR,
-					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-					 errmsg("cannot call %s on an array",
-							"jsonb_object_keys")));
-
-		funcctx = SRF_FIRSTCALL_INIT();
-		oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
-
-		state = palloc(sizeof(OkeysState));
-
-		state->result_size = JB_ROOT_COUNT(jb);
-		state->result_count = 0;
-		state->sent_count = 0;
-		state->result = palloc(state->result_size * sizeof(char *));
-
-		it = JsonbIteratorInit(&jb->root);
-
-		while ((r = JsonbIteratorNext(&it, &v, skipNested)) != WJB_DONE)
-		{
-			skipNested = true;
-
-			if (r == WJB_KEY)
-			{
-				char	   *cstr;
-
-				cstr = palloc(v.val.string.len + 1 * sizeof(char));
-				memcpy(cstr, v.val.string.val, v.val.string.len);
-				cstr[v.val.string.len] = '\0';
-				state->result[state->result_count++] = cstr;
-			}
-		}
-
-		MemoryContextSwitchTo(oldcontext);
-		funcctx->user_fctx = (void *) state;
-	}
-
-	funcctx = SRF_PERCALL_SETUP();
-	state = (OkeysState *) funcctx->user_fctx;
-
-	if (state->sent_count < state->result_count)
-	{
-		char	   *nxt = state->result[state->sent_count++];
-
-		SRF_RETURN_NEXT(funcctx, CStringGetTextDatum(nxt));
-	}
-
-	SRF_RETURN_DONE(funcctx);
+	elog(ERROR, "jsonb_object_keys is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
