diff --git a/src/backend/commands/sequence.c b/src/backend/commands/sequence.c
index d5aff5b..f6e7e3d 100644
--- a/src/backend/commands/sequence.c
+++ b/src/backend/commands/sequence.c
@@ -896,36 +896,8 @@ currval_oid(PG_FUNCTION_ARGS)
 Datum
 lastval(PG_FUNCTION_ARGS)
 {
-	Relation	seqrel;
-	int64		result;
-
-	if (last_used_seq == NULL)
-		ereport(ERROR,
-				(errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
-				 errmsg("lastval is not yet defined in this session")));
-
-	/* Someone may have dropped the sequence since the last nextval() */
-	if (!SearchSysCacheExists1(RELOID, ObjectIdGetDatum(last_used_seq->relid)))
-		ereport(ERROR,
-				(errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
-				 errmsg("lastval is not yet defined in this session")));
-
-	seqrel = lock_and_open_sequence(last_used_seq);
-
-	/* nextval() must have already been called for this sequence */
-	Assert(last_used_seq->last_valid);
-
-	if (pg_class_aclcheck(last_used_seq->relid, GetUserId(),
-						  ACL_SELECT | ACL_USAGE) != ACLCHECK_OK)
-		ereport(ERROR,
-				(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
-				 errmsg("permission denied for sequence %s",
-						RelationGetRelationName(seqrel))));
-
-	result = last_used_seq->last;
-	sequence_close(seqrel, NoLock);
-
-	PG_RETURN_INT64(result);
+	elog(ERROR, "lastval is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
