diff --git a/src/backend/utils/adt/timestamp.c b/src/backend/utils/adt/timestamp.c
index b57739b..a9ada67 100644
--- a/src/backend/utils/adt/timestamp.c
+++ b/src/backend/utils/adt/timestamp.c
@@ -667,64 +667,18 @@ make_timestamp(PG_FUNCTION_ARGS)
 	PG_RETURN_TIMESTAMP(result);
 }
 
-/*
- * make_timestamptz() - timestamp with time zone constructor
- */
 Datum
 make_timestamptz(PG_FUNCTION_ARGS)
 {
-	int32		year = PG_GETARG_INT32(0);
-	int32		month = PG_GETARG_INT32(1);
-	int32		mday = PG_GETARG_INT32(2);
-	int32		hour = PG_GETARG_INT32(3);
-	int32		min = PG_GETARG_INT32(4);
-	float8		sec = PG_GETARG_FLOAT8(5);
-	Timestamp	result;
-
-	result = make_timestamp_internal(year, month, mday,
-									 hour, min, sec);
-
-	PG_RETURN_TIMESTAMPTZ(timestamp2timestamptz(result));
+	elog(ERROR, "make_timestamptz is not implemented");
+	PG_RETURN_NULL();
 }
 
-/*
- * Construct a timestamp with time zone.
- *		As above, but the time zone is specified as seventh argument.
- */
 Datum
 make_timestamptz_at_timezone(PG_FUNCTION_ARGS)
 {
-	int32		year = PG_GETARG_INT32(0);
-	int32		month = PG_GETARG_INT32(1);
-	int32		mday = PG_GETARG_INT32(2);
-	int32		hour = PG_GETARG_INT32(3);
-	int32		min = PG_GETARG_INT32(4);
-	float8		sec = PG_GETARG_FLOAT8(5);
-	text	   *zone = PG_GETARG_TEXT_PP(6);
-	TimestampTz result;
-	Timestamp	timestamp;
-	struct pg_tm tt;
-	int			tz;
-	fsec_t		fsec;
-
-	timestamp = make_timestamp_internal(year, month, mday,
-										hour, min, sec);
-
-	if (timestamp2tm(timestamp, NULL, &tt, &fsec, NULL, NULL) != 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("timestamp out of range")));
-
-	tz = parse_sane_timezone(&tt, zone);
-
-	result = dt2local(timestamp, -tz);
-
-	if (!IS_VALID_TIMESTAMP(result))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("timestamp out of range")));
-
-	PG_RETURN_TIMESTAMPTZ(result);
+	elog(ERROR, "make_timestamptz_at_timezone is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
