diff --git a/src/backend/utils/adt/network.c b/src/backend/utils/adt/network.c
index 640fc37..3a8cbc1 100644
--- a/src/backend/utils/adt/network.c
+++ b/src/backend/utils/adt/network.c
@@ -1329,45 +1329,8 @@ network_broadcast(PG_FUNCTION_ARGS)
 Datum
 network_network(PG_FUNCTION_ARGS)
 {
-	inet	   *ip = PG_GETARG_INET_PP(0);
-	inet	   *dst;
-	int			byte;
-	int			bits;
-	unsigned char mask;
-	unsigned char *a,
-			   *b;
-
-	/* make sure any unused bits are zeroed */
-	dst = (inet *) palloc0(sizeof(inet));
-
-	bits = ip_bits(ip);
-	a = ip_addr(ip);
-	b = ip_addr(dst);
-
-	byte = 0;
-
-	while (bits)
-	{
-		if (bits >= 8)
-		{
-			mask = 0xff;
-			bits -= 8;
-		}
-		else
-		{
-			mask = 0xff << (8 - bits);
-			bits = 0;
-		}
-
-		b[byte] = a[byte] & mask;
-		byte++;
-	}
-
-	ip_family(dst) = ip_family(ip);
-	ip_bits(dst) = ip_bits(ip);
-	SET_INET_VARSIZE(dst);
-
-	PG_RETURN_INET_P(dst);
+	elog(ERROR, "network_network is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
