diff --git a/src/backend/utils/adt/misc.c b/src/backend/utils/adt/misc.c
index 0e6c458..859e130 100644
--- a/src/backend/utils/adt/misc.c
+++ b/src/backend/utils/adt/misc.c
@@ -852,143 +852,11 @@ is_ident_cont(unsigned char c)
 	return is_ident_start(c);
 }
 
-/*
- * parse_ident - parse a SQL qualified identifier into separate identifiers.
- * When strict mode is active (second parameter), then any chars after
- * the last identifier are disallowed.
- */
 Datum
 parse_ident(PG_FUNCTION_ARGS)
 {
-	text	   *qualname = PG_GETARG_TEXT_PP(0);
-	bool		strict = PG_GETARG_BOOL(1);
-	char	   *qualname_str = text_to_cstring(qualname);
-	ArrayBuildState *astate = NULL;
-	char	   *nextp;
-	bool		after_dot = false;
-
-	/*
-	 * The code below scribbles on qualname_str in some cases, so we should
-	 * reconvert qualname if we need to show the original string in error
-	 * messages.
-	 */
-	nextp = qualname_str;
-
-	/* skip leading whitespace */
-	while (scanner_isspace(*nextp))
-		nextp++;
-
-	for (;;)
-	{
-		char	   *curname;
-		bool		missing_ident = true;
-
-		if (*nextp == '"')
-		{
-			char	   *endp;
-
-			curname = nextp + 1;
-			for (;;)
-			{
-				endp = strchr(nextp + 1, '"');
-				if (endp == NULL)
-					ereport(ERROR,
-							(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-							 errmsg("string is not a valid identifier: \"%s\"",
-									text_to_cstring(qualname)),
-							 errdetail("String has unclosed double quotes.")));
-				if (endp[1] != '"')
-					break;
-				memmove(endp, endp + 1, strlen(endp));
-				nextp = endp;
-			}
-			nextp = endp + 1;
-			*endp = '\0';
-
-			if (endp - curname == 0)
-				ereport(ERROR,
-						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-						 errmsg("string is not a valid identifier: \"%s\"",
-								text_to_cstring(qualname)),
-						 errdetail("Quoted identifier must not be empty.")));
-
-			astate = accumArrayResult(astate, CStringGetTextDatum(curname),
-									  false, TEXTOID, CurrentMemoryContext);
-			missing_ident = false;
-		}
-		else if (is_ident_start((unsigned char) *nextp))
-		{
-			char	   *downname;
-			int			len;
-			text	   *part;
-
-			curname = nextp++;
-			while (is_ident_cont((unsigned char) *nextp))
-				nextp++;
-
-			len = nextp - curname;
-
-			/*
-			 * We don't implicitly truncate identifiers. This is useful for
-			 * allowing the user to check for specific parts of the identifier
-			 * being too long. It's easy enough for the user to get the
-			 * truncated names by casting our output to name[].
-			 */
-			downname = downcase_identifier(curname, len, false, false);
-			part = cstring_to_text_with_len(downname, len);
-			astate = accumArrayResult(astate, PointerGetDatum(part), false,
-									  TEXTOID, CurrentMemoryContext);
-			missing_ident = false;
-		}
-
-		if (missing_ident)
-		{
-			/* Different error messages based on where we failed. */
-			if (*nextp == '.')
-				ereport(ERROR,
-						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-						 errmsg("string is not a valid identifier: \"%s\"",
-								text_to_cstring(qualname)),
-						 errdetail("No valid identifier before \".\".")));
-			else if (after_dot)
-				ereport(ERROR,
-						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-						 errmsg("string is not a valid identifier: \"%s\"",
-								text_to_cstring(qualname)),
-						 errdetail("No valid identifier after \".\".")));
-			else
-				ereport(ERROR,
-						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-						 errmsg("string is not a valid identifier: \"%s\"",
-								text_to_cstring(qualname))));
-		}
-
-		while (scanner_isspace(*nextp))
-			nextp++;
-
-		if (*nextp == '.')
-		{
-			after_dot = true;
-			nextp++;
-			while (scanner_isspace(*nextp))
-				nextp++;
-		}
-		else if (*nextp == '\0')
-		{
-			break;
-		}
-		else
-		{
-			if (strict)
-				ereport(ERROR,
-						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-						 errmsg("string is not a valid identifier: \"%s\"",
-								text_to_cstring(qualname))));
-			break;
-		}
-	}
-
-	PG_RETURN_DATUM(makeArrayResult(astate, CurrentMemoryContext));
+	elog(ERROR, "parse_ident is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
