diff --git a/src/backend/utils/adt/varlena.c b/src/backend/utils/adt/varlena.c
index 15d2cab..671f949 100644
--- a/src/backend/utils/adt/varlena.c
+++ b/src/backend/utils/adt/varlena.c
@@ -5053,52 +5053,11 @@ to_hex64(PG_FUNCTION_ARGS)
 	PG_RETURN_TEXT_P(convert_to_base(value, 16));
 }
 
-/*
- * Return the size of a datum, possibly compressed
- *
- * Works on any data type
- */
 Datum
 pg_column_size(PG_FUNCTION_ARGS)
 {
-	Datum		value = PG_GETARG_DATUM(0);
-	int32		result;
-	int			typlen;
-
-	/* On first call, get the input type's typlen, and save at *fn_extra */
-	if (fcinfo->flinfo->fn_extra == NULL)
-	{
-		/* Lookup the datatype of the supplied argument */
-		Oid			argtypeid = get_fn_expr_argtype(fcinfo->flinfo, 0);
-
-		typlen = get_typlen(argtypeid);
-		if (typlen == 0)		/* should not happen */
-			elog(ERROR, "cache lookup failed for type %u", argtypeid);
-
-		fcinfo->flinfo->fn_extra = MemoryContextAlloc(fcinfo->flinfo->fn_mcxt,
-													  sizeof(int));
-		*((int *) fcinfo->flinfo->fn_extra) = typlen;
-	}
-	else
-		typlen = *((int *) fcinfo->flinfo->fn_extra);
-
-	if (typlen == -1)
-	{
-		/* varlena type, possibly toasted */
-		result = toast_datum_size(value);
-	}
-	else if (typlen == -2)
-	{
-		/* cstring */
-		result = strlen(DatumGetCString(value)) + 1;
-	}
-	else
-	{
-		/* ordinary fixed-width type */
-		result = typlen;
-	}
-
-	PG_RETURN_INT32(result);
+	elog(ERROR, "pg_column_size is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
