diff --git a/src/backend/utils/adt/network.c b/src/backend/utils/adt/network.c
index 640fc37..3e6201e 100644
--- a/src/backend/utils/adt/network.c
+++ b/src/backend/utils/adt/network.c
@@ -323,42 +323,15 @@ inet_to_cidr(PG_FUNCTION_ARGS)
 Datum
 inet_set_masklen(PG_FUNCTION_ARGS)
 {
-	inet	   *src = PG_GETARG_INET_PP(0);
-	int			bits = PG_GETARG_INT32(1);
-	inet	   *dst;
-
-	if (bits == -1)
-		bits = ip_maxbits(src);
-
-	if ((bits < 0) || (bits > ip_maxbits(src)))
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("invalid mask length: %d", bits)));
-
-	/* clone the original data */
-	dst = (inet *) palloc(VARSIZE_ANY(src));
-	memcpy(dst, src, VARSIZE_ANY(src));
-
-	ip_bits(dst) = bits;
-
-	PG_RETURN_INET_P(dst);
+	elog(ERROR, "inet_set_masklen is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
 cidr_set_masklen(PG_FUNCTION_ARGS)
 {
-	inet	   *src = PG_GETARG_INET_PP(0);
-	int			bits = PG_GETARG_INT32(1);
-
-	if (bits == -1)
-		bits = ip_maxbits(src);
-
-	if ((bits < 0) || (bits > ip_maxbits(src)))
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("invalid mask length: %d", bits)));
-
-	PG_RETURN_INET_P(cidr_set_masklen_internal(src, bits));
+	elog(ERROR, "cidr_set_masklen is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
