diff --git a/src/backend/utils/adt/geo_ops.c b/src/backend/utils/adt/geo_ops.c
index 07d1649..b75ce3f 100644
--- a/src/backend/utils/adt/geo_ops.c
+++ b/src/backend/utils/adt/geo_ops.c
@@ -2007,10 +2007,8 @@ point_dt(Point *pt1, Point *pt2)
 Datum
 point_slope(PG_FUNCTION_ARGS)
 {
-	Point	   *pt1 = PG_GETARG_POINT_P(0);
-	Point	   *pt2 = PG_GETARG_POINT_P(1);
-
-	PG_RETURN_FLOAT8(point_sl(pt1, pt2));
+	elog(ERROR, "point_slope is not implemented");
+	PG_RETURN_NULL();
 }
 
 
