diff --git a/src/backend/utils/adt/varlena.c b/src/backend/utils/adt/varlena.c
index 15d2cab..bed4713 100644
--- a/src/backend/utils/adt/varlena.c
+++ b/src/backend/utils/adt/varlena.c
@@ -4404,138 +4404,11 @@ replace_text_regexp(text *src_text, text *pattern_text,
 	return ret_text;
 }
 
-/*
- * split_part
- * parse input string based on provided field separator
- * return N'th item (1 based, negative counts from end)
- */
 Datum
 split_part(PG_FUNCTION_ARGS)
 {
-	text	   *inputstring = PG_GETARG_TEXT_PP(0);
-	text	   *fldsep = PG_GETARG_TEXT_PP(1);
-	int			fldnum = PG_GETARG_INT32(2);
-	int			inputstring_len;
-	int			fldsep_len;
-	TextPositionState state;
-	char	   *start_ptr;
-	char	   *end_ptr;
-	text	   *result_text;
-	bool		found;
-
-	/* field number is 1 based */
-	if (fldnum == 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("field position must not be zero")));
-
-	inputstring_len = VARSIZE_ANY_EXHDR(inputstring);
-	fldsep_len = VARSIZE_ANY_EXHDR(fldsep);
-
-	/* return empty string for empty input string */
-	if (inputstring_len < 1)
-		PG_RETURN_TEXT_P(cstring_to_text(""));
-
-	/* handle empty field separator */
-	if (fldsep_len < 1)
-	{
-		/* if first or last field, return input string, else empty string */
-		if (fldnum == 1 || fldnum == -1)
-			PG_RETURN_TEXT_P(inputstring);
-		else
-			PG_RETURN_TEXT_P(cstring_to_text(""));
-	}
-
-	/* find the first field separator */
-	text_position_setup(inputstring, fldsep, PG_GET_COLLATION(), &state);
-
-	found = text_position_next(&state);
-
-	/* special case if fldsep not found at all */
-	if (!found)
-	{
-		text_position_cleanup(&state);
-		/* if first or last field, return input string, else empty string */
-		if (fldnum == 1 || fldnum == -1)
-			PG_RETURN_TEXT_P(inputstring);
-		else
-			PG_RETURN_TEXT_P(cstring_to_text(""));
-	}
-
-	/*
-	 * take care of a negative field number (i.e. count from the right) by
-	 * converting to a positive field number; we need total number of fields
-	 */
-	if (fldnum < 0)
-	{
-		/* we found a fldsep, so there are at least two fields */
-		int			numfields = 2;
-
-		while (text_position_next(&state))
-			numfields++;
-
-		/* special case of last field does not require an extra pass */
-		if (fldnum == -1)
-		{
-			start_ptr = text_position_get_match_ptr(&state) + fldsep_len;
-			end_ptr = VARDATA_ANY(inputstring) + inputstring_len;
-			text_position_cleanup(&state);
-			PG_RETURN_TEXT_P(cstring_to_text_with_len(start_ptr,
-													  end_ptr - start_ptr));
-		}
-
-		/* else, convert fldnum to positive notation */
-		fldnum += numfields + 1;
-
-		/* if nonexistent field, return empty string */
-		if (fldnum <= 0)
-		{
-			text_position_cleanup(&state);
-			PG_RETURN_TEXT_P(cstring_to_text(""));
-		}
-
-		/* reset to pointing at first match, but now with positive fldnum */
-		text_position_reset(&state);
-		found = text_position_next(&state);
-		Assert(found);
-	}
-
-	/* identify bounds of first field */
-	start_ptr = VARDATA_ANY(inputstring);
-	end_ptr = text_position_get_match_ptr(&state);
-
-	while (found && --fldnum > 0)
-	{
-		/* identify bounds of next field */
-		start_ptr = end_ptr + fldsep_len;
-		found = text_position_next(&state);
-		if (found)
-			end_ptr = text_position_get_match_ptr(&state);
-	}
-
-	text_position_cleanup(&state);
-
-	if (fldnum > 0)
-	{
-		/* N'th field separator not found */
-		/* if last field requested, return it, else empty string */
-		if (fldnum == 1)
-		{
-			int			last_len = start_ptr - VARDATA_ANY(inputstring);
-
-			result_text = cstring_to_text_with_len(start_ptr,
-												   inputstring_len - last_len);
-		}
-		else
-			result_text = cstring_to_text("");
-	}
-	else
-	{
-		/* non-last field requested */
-		result_text = cstring_to_text_with_len(start_ptr, end_ptr - start_ptr);
-	}
-
-	PG_RETURN_TEXT_P(result_text);
+	elog(ERROR, "split_part is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
