diff --git a/src/backend/utils/adt/timestamp.c b/src/backend/utils/adt/timestamp.c
index b57739b..447d4bd 100644
--- a/src/backend/utils/adt/timestamp.c
+++ b/src/backend/utils/adt/timestamp.c
@@ -409,73 +409,11 @@ AdjustTimestampForTypmod(Timestamp *time, int32 typmod, Node *escontext)
 	return true;
 }
 
-/* timestamptz_in()
- * Convert a string to internal form.
- */
 Datum
 timestamptz_in(PG_FUNCTION_ARGS)
 {
-	char	   *str = PG_GETARG_CSTRING(0);
-#ifdef NOT_USED
-	Oid			typelem = PG_GETARG_OID(1);
-#endif
-	int32		typmod = PG_GETARG_INT32(2);
-	Node	   *escontext = fcinfo->context;
-	TimestampTz result;
-	fsec_t		fsec;
-	struct pg_tm tt,
-			   *tm = &tt;
-	int			tz;
-	int			dtype;
-	int			nf;
-	int			dterr;
-	char	   *field[MAXDATEFIELDS];
-	int			ftype[MAXDATEFIELDS];
-	char		workbuf[MAXDATELEN + MAXDATEFIELDS];
-	DateTimeErrorExtra extra;
-
-	dterr = ParseDateTime(str, workbuf, sizeof(workbuf),
-						  field, ftype, MAXDATEFIELDS, &nf);
-	if (dterr == 0)
-		dterr = DecodeDateTime(field, ftype, nf,
-							   &dtype, tm, &fsec, &tz, &extra);
-	if (dterr != 0)
-	{
-		DateTimeParseError(dterr, &extra, str, "timestamp with time zone",
-						   escontext);
-		PG_RETURN_NULL();
-	}
-
-	switch (dtype)
-	{
-		case DTK_DATE:
-			if (tm2timestamp(tm, fsec, &tz, &result) != 0)
-				ereturn(escontext, (Datum) 0,
-						(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-						 errmsg("timestamp out of range: \"%s\"", str)));
-			break;
-
-		case DTK_EPOCH:
-			result = SetEpochTimestamp();
-			break;
-
-		case DTK_LATE:
-			TIMESTAMP_NOEND(result);
-			break;
-
-		case DTK_EARLY:
-			TIMESTAMP_NOBEGIN(result);
-			break;
-
-		default:
-			elog(ERROR, "unexpected dtype %d while parsing timestamptz \"%s\"",
-				 dtype, str);
-			TIMESTAMP_NOEND(result);
-	}
-
-	AdjustTimestampForTypmod(&result, typmod, escontext);
-
-	PG_RETURN_TIMESTAMPTZ(result);
+	elog(ERROR, "timestamptz_in is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
