diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..9c4b12e 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -2328,40 +2328,11 @@ timetz_out(PG_FUNCTION_ARGS)
 	PG_RETURN_CSTRING(result);
 }
 
-/*
- *		timetz_recv			- converts external binary format to timetz
- */
 Datum
 timetz_recv(PG_FUNCTION_ARGS)
 {
-	StringInfo	buf = (StringInfo) PG_GETARG_POINTER(0);
-
-#ifdef NOT_USED
-	Oid			typelem = PG_GETARG_OID(1);
-#endif
-	int32		typmod = PG_GETARG_INT32(2);
-	TimeTzADT  *result;
-
-	result = (TimeTzADT *) palloc(sizeof(TimeTzADT));
-
-	result->time = pq_getmsgint64(buf);
-
-	if (result->time < INT64CONST(0) || result->time > USECS_PER_DAY)
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("time out of range")));
-
-	result->zone = pq_getmsgint(buf, sizeof(result->zone));
-
-	/* Check for sane GMT displacement; see notes in datatype/timestamp.h */
-	if (result->zone <= -TZDISP_LIMIT || result->zone >= TZDISP_LIMIT)
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_TIME_ZONE_DISPLACEMENT_VALUE),
-				 errmsg("time zone displacement out of range")));
-
-	AdjustTimeForTypmod(&(result->time), typmod);
-
-	PG_RETURN_TIMETZADT_P(result);
+	elog(ERROR, "timetz_recv is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
