diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..32af3e6 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -960,10 +960,8 @@ timestamp_ge_date(PG_FUNCTION_ARGS)
 Datum
 timestamp_cmp_date(PG_FUNCTION_ARGS)
 {
-	Timestamp	dt1 = PG_GETARG_TIMESTAMP(0);
-	DateADT		dateVal = PG_GETARG_DATEADT(1);
-
-	PG_RETURN_INT32(-date_cmp_timestamp_internal(dateVal, dt1));
+	elog(ERROR, "timestamp_cmp_date is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
