diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..e4bf06e 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -1023,10 +1023,8 @@ timestamptz_ge_date(PG_FUNCTION_ARGS)
 Datum
 timestamptz_cmp_date(PG_FUNCTION_ARGS)
 {
-	TimestampTz dt1 = PG_GETARG_TIMESTAMPTZ(0);
-	DateADT		dateVal = PG_GETARG_DATEADT(1);
-
-	PG_RETURN_INT32(-date_cmp_timestamptz_internal(dateVal, dt1));
+	elog(ERROR, "timestamptz_cmp_date is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
