diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..ef94055 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -2045,27 +2045,11 @@ time_mi_time(PG_FUNCTION_ARGS)
 	PG_RETURN_INTERVAL_P(result);
 }
 
-/* time_pl_interval()
- * Add interval to time.
- */
 Datum
 time_pl_interval(PG_FUNCTION_ARGS)
 {
-	TimeADT		time = PG_GETARG_TIMEADT(0);
-	Interval   *span = PG_GETARG_INTERVAL_P(1);
-	TimeADT		result;
-
-	if (INTERVAL_NOT_FINITE(span))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("cannot add infinite interval to time")));
-
-	result = time + span->time;
-	result -= result / USECS_PER_DAY * USECS_PER_DAY;
-	if (result < INT64CONST(0))
-		result += USECS_PER_DAY;
-
-	PG_RETURN_TIMEADT(result);
+	elog(ERROR, "time_pl_interval is not implemented");
+	PG_RETURN_NULL();
 }
 
 /* time_mi_interval()
