diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..34176fe 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -906,10 +906,8 @@ date_cmp_timestamptz(PG_FUNCTION_ARGS)
 Datum
 timestamp_eq_date(PG_FUNCTION_ARGS)
 {
-	Timestamp	dt1 = PG_GETARG_TIMESTAMP(0);
-	DateADT		dateVal = PG_GETARG_DATEADT(1);
-
-	PG_RETURN_BOOL(date_cmp_timestamp_internal(dateVal, dt1) == 0);
+	elog(ERROR, "timestamp_eq_date is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
