diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..241b373 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -1767,10 +1767,8 @@ time_larger(PG_FUNCTION_ARGS)
 Datum
 time_smaller(PG_FUNCTION_ARGS)
 {
-	TimeADT		time1 = PG_GETARG_TIMEADT(0);
-	TimeADT		time2 = PG_GETARG_TIMEADT(1);
-
-	PG_RETURN_TIMEADT((time1 < time2) ? time1 : time2);
+	elog(ERROR, "time_smaller is not implemented");
+	PG_RETURN_NULL();
 }
 
 /* overlaps_time() --- implements the SQL OVERLAPS operator.
