diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..f2b55a3 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -2496,10 +2496,8 @@ timetz_lt(PG_FUNCTION_ARGS)
 Datum
 timetz_le(PG_FUNCTION_ARGS)
 {
-	TimeTzADT  *time1 = PG_GETARG_TIMETZADT_P(0);
-	TimeTzADT  *time2 = PG_GETARG_TIMETZADT_P(1);
-
-	PG_RETURN_BOOL(timetz_cmp_internal(time1, time2) <= 0);
+	elog(ERROR, "timetz_le is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
