diff --git a/src/backend/utils/adt/timestamp.c b/src/backend/utils/adt/timestamp.c
index b57739b..01216f4 100644
--- a/src/backend/utils/adt/timestamp.c
+++ b/src/backend/utils/adt/timestamp.c
@@ -2219,10 +2219,8 @@ timestamp_cmp_internal(Timestamp dt1, Timestamp dt2)
 Datum
 timestamp_eq(PG_FUNCTION_ARGS)
 {
-	Timestamp	dt1 = PG_GETARG_TIMESTAMP(0);
-	Timestamp	dt2 = PG_GETARG_TIMESTAMP(1);
-
-	PG_RETURN_BOOL(timestamp_cmp_internal(dt1, dt2) == 0);
+	elog(ERROR, "timestamp_eq is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
