diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..6d114d0 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -924,10 +924,8 @@ timestamp_ne_date(PG_FUNCTION_ARGS)
 Datum
 timestamp_lt_date(PG_FUNCTION_ARGS)
 {
-	Timestamp	dt1 = PG_GETARG_TIMESTAMP(0);
-	DateADT		dateVal = PG_GETARG_DATEADT(1);
-
-	PG_RETURN_BOOL(date_cmp_timestamp_internal(dateVal, dt1) > 0);
+	elog(ERROR, "timestamp_lt_date is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
