diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 9c854e0..03bd7d5 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -2564,15 +2564,8 @@ timetz_hash_extended(PG_FUNCTION_ARGS)
 Datum
 timetz_larger(PG_FUNCTION_ARGS)
 {
-	TimeTzADT  *time1 = PG_GETARG_TIMETZADT_P(0);
-	TimeTzADT  *time2 = PG_GETARG_TIMETZADT_P(1);
-	TimeTzADT  *result;
-
-	if (timetz_cmp_internal(time1, time2) > 0)
-		result = time1;
-	else
-		result = time2;
-	PG_RETURN_TIMETZADT_P(result);
+	elog(ERROR, "timetz_larger is not implemented");
+	PG_RETURN_NULL();
 }
 
 Datum
