diff --git a/src/backend/utils/adt/formatting.c b/src/backend/utils/adt/formatting.c
index d918350..0b2d777 100644
--- a/src/backend/utils/adt/formatting.c
+++ b/src/backend/utils/adt/formatting.c
@@ -6487,63 +6487,11 @@ do { \
 	SET_VARSIZE(result, len + VARHDRSZ); \
 } while (0)
 
-/* -------------------
- * NUMERIC to_number() (convert string to numeric)
- * -------------------
- */
 Datum
 numeric_to_number(PG_FUNCTION_ARGS)
 {
-	text	   *value = PG_GETARG_TEXT_PP(0);
-	text	   *fmt = PG_GETARG_TEXT_PP(1);
-	NUMDesc		Num;
-	Datum		result;
-	FormatNode *format;
-	char	   *numstr;
-	bool		shouldFree;
-	int			len = 0;
-	int			scale,
-				precision;
-
-	len = VARSIZE_ANY_EXHDR(fmt);
-
-	if (len <= 0 || len >= INT_MAX / NUM_MAX_ITEM_SIZ)
-		PG_RETURN_NULL();
-
-	format = NUM_cache(len, &Num, fmt, &shouldFree);
-
-	numstr = (char *) palloc((len * NUM_MAX_ITEM_SIZ) + 1);
-
-	NUM_processor(format, &Num, VARDATA_ANY(value), numstr,
-				  VARSIZE_ANY_EXHDR(value), 0, 0, false, PG_GET_COLLATION());
-
-	scale = Num.post;
-	precision = Num.pre + Num.multi + scale;
-
-	if (shouldFree)
-		pfree(format);
-
-	result = DirectFunctionCall3(numeric_in,
-								 CStringGetDatum(numstr),
-								 ObjectIdGetDatum(InvalidOid),
-								 Int32GetDatum(((precision << 16) | scale) + VARHDRSZ));
-
-	if (IS_MULTI(&Num))
-	{
-		Numeric		x;
-		Numeric		a = int64_to_numeric(10);
-		Numeric		b = int64_to_numeric(-Num.multi);
-
-		x = DatumGetNumeric(DirectFunctionCall2(numeric_power,
-												NumericGetDatum(a),
-												NumericGetDatum(b)));
-		result = DirectFunctionCall2(numeric_mul,
-									 result,
-									 NumericGetDatum(x));
-	}
-
-	pfree(numstr);
-	return result;
+	elog(ERROR, "numeric_to_number is not implemented");
+	PG_RETURN_NULL();
 }
 
 /* ------------------
