diff --git a/src/backend/utils/adt/formatting.c b/src/backend/utils/adt/formatting.c
index d918350..978b4ee 100644
--- a/src/backend/utils/adt/formatting.c
+++ b/src/backend/utils/adt/formatting.c
@@ -4483,45 +4483,11 @@ interval_to_char(PG_FUNCTION_ARGS)
 	PG_RETURN_TEXT_P(res);
 }
 
-/* ---------------------
- * TO_TIMESTAMP()
- *
- * Make Timestamp from date_str which is formatted at argument 'fmt'
- * ( to_timestamp is reverse to_char() )
- * ---------------------
- */
 Datum
 to_timestamp(PG_FUNCTION_ARGS)
 {
-	text	   *date_txt = PG_GETARG_TEXT_PP(0);
-	text	   *fmt = PG_GETARG_TEXT_PP(1);
-	Oid			collid = PG_GET_COLLATION();
-	Timestamp	result;
-	int			tz;
-	struct pg_tm tm;
-	struct fmt_tz ftz;
-	fsec_t		fsec;
-	int			fprec;
-
-	do_to_timestamp(date_txt, fmt, collid, false,
-					&tm, &fsec, &ftz, &fprec, NULL, NULL);
-
-	/* Use the specified time zone, if any. */
-	if (ftz.has_tz)
-		tz = ftz.gmtoffset;
-	else
-		tz = DetermineTimeZoneOffset(&tm, session_timezone);
-
-	if (tm2timestamp(&tm, fsec, &tz, &result) != 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("timestamp out of range")));
-
-	/* Use the specified fractional precision, if any. */
-	if (fprec)
-		AdjustTimestampForTypmod(&result, fprec, NULL);
-
-	PG_RETURN_TIMESTAMP(result);
+	elog(ERROR, "to_timestamp is not implemented");
+	PG_RETURN_NULL();
 }
 
 /* ----------
diff --git a/src/backend/utils/adt/timestamp.c b/src/backend/utils/adt/timestamp.c
index b57739b..4ffdff4 100644
--- a/src/backend/utils/adt/timestamp.c
+++ b/src/backend/utils/adt/timestamp.c
@@ -727,55 +727,11 @@ make_timestamptz_at_timezone(PG_FUNCTION_ARGS)
 	PG_RETURN_TIMESTAMPTZ(result);
 }
 
-/*
- * to_timestamp(double precision)
- * Convert UNIX epoch to timestamptz.
- */
 Datum
 float8_timestamptz(PG_FUNCTION_ARGS)
 {
-	float8		seconds = PG_GETARG_FLOAT8(0);
-	TimestampTz result;
-
-	/* Deal with NaN and infinite inputs ... */
-	if (isnan(seconds))
-		ereport(ERROR,
-				(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-				 errmsg("timestamp cannot be NaN")));
-
-	if (isinf(seconds))
-	{
-		if (seconds < 0)
-			TIMESTAMP_NOBEGIN(result);
-		else
-			TIMESTAMP_NOEND(result);
-	}
-	else
-	{
-		/* Out of range? */
-		if (seconds <
-			(float8) SECS_PER_DAY * (DATETIME_MIN_JULIAN - UNIX_EPOCH_JDATE)
-			|| seconds >=
-			(float8) SECS_PER_DAY * (TIMESTAMP_END_JULIAN - UNIX_EPOCH_JDATE))
-			ereport(ERROR,
-					(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-					 errmsg("timestamp out of range: \"%g\"", seconds)));
-
-		/* Convert UNIX epoch to Postgres epoch */
-		seconds -= ((POSTGRES_EPOCH_JDATE - UNIX_EPOCH_JDATE) * SECS_PER_DAY);
-
-		seconds = rint(seconds * USECS_PER_SEC);
-		result = (int64) seconds;
-
-		/* Recheck in case roundoff produces something just out of range */
-		if (!IS_VALID_TIMESTAMP(result))
-			ereport(ERROR,
-					(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
-					 errmsg("timestamp out of range: \"%g\"",
-							PG_GETARG_FLOAT8(0))));
-	}
-
-	PG_RETURN_TIMESTAMP(result);
+	elog(ERROR, "float8_timestamptz is not implemented");
+	PG_RETURN_NULL();
 }
 
 /* timestamptz_out()
