diff --git a/src/backend/utils/adt/arrayfuncs.c b/src/backend/utils/adt/arrayfuncs.c
index 75cc4ee..66b7d7c 100644
--- a/src/backend/utils/adt/arrayfuncs.c
+++ b/src/backend/utils/adt/arrayfuncs.c
@@ -6679,72 +6679,11 @@ array_replace(PG_FUNCTION_ARGS)
 	PG_RETURN_ARRAYTYPE_P(array);
 }
 
-/*
- * Implements width_bucket(anyelement, anyarray).
- *
- * 'thresholds' is an array containing lower bound values for each bucket;
- * these must be sorted from smallest to largest, or bogus results will be
- * produced.  If N thresholds are supplied, the output is from 0 to N:
- * 0 is for inputs < first threshold, N is for inputs >= last threshold.
- */
 Datum
 width_bucket_array(PG_FUNCTION_ARGS)
 {
-	Datum		operand = PG_GETARG_DATUM(0);
-	ArrayType  *thresholds = PG_GETARG_ARRAYTYPE_P(1);
-	Oid			collation = PG_GET_COLLATION();
-	Oid			element_type = ARR_ELEMTYPE(thresholds);
-	int			result;
-
-	/* Check input */
-	if (ARR_NDIM(thresholds) > 1)
-		ereport(ERROR,
-				(errcode(ERRCODE_ARRAY_SUBSCRIPT_ERROR),
-				 errmsg("thresholds must be one-dimensional array")));
-
-	if (array_contains_nulls(thresholds))
-		ereport(ERROR,
-				(errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED),
-				 errmsg("thresholds array must not contain NULLs")));
-
-	/* We have a dedicated implementation for float8 data */
-	if (element_type == FLOAT8OID)
-		result = width_bucket_array_float8(operand, thresholds);
-	else
-	{
-		TypeCacheEntry *typentry;
-
-		/* Cache information about the input type */
-		typentry = (TypeCacheEntry *) fcinfo->flinfo->fn_extra;
-		if (typentry == NULL ||
-			typentry->type_id != element_type)
-		{
-			typentry = lookup_type_cache(element_type,
-										 TYPECACHE_CMP_PROC_FINFO);
-			if (!OidIsValid(typentry->cmp_proc_finfo.fn_oid))
-				ereport(ERROR,
-						(errcode(ERRCODE_UNDEFINED_FUNCTION),
-						 errmsg("could not identify a comparison function for type %s",
-								format_type_be(element_type))));
-			fcinfo->flinfo->fn_extra = (void *) typentry;
-		}
-
-		/*
-		 * We have separate implementation paths for fixed- and variable-width
-		 * types, since indexing the array is a lot cheaper in the first case.
-		 */
-		if (typentry->typlen > 0)
-			result = width_bucket_array_fixed(operand, thresholds,
-											  collation, typentry);
-		else
-			result = width_bucket_array_variable(operand, thresholds,
-												 collation, typentry);
-	}
-
-	/* Avoid leaking memory when handed toasted input. */
-	PG_FREE_IF_COPY(thresholds, 1);
-
-	PG_RETURN_INT32(result);
+	elog(ERROR, "width_bucket_array is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
diff --git a/src/backend/utils/adt/float.c b/src/backend/utils/adt/float.c
index 22421cb..eb73c18 100644
--- a/src/backend/utils/adt/float.c
+++ b/src/backend/utils/adt/float.c
@@ -3971,112 +3971,10 @@ float84ge(PG_FUNCTION_ARGS)
 	PG_RETURN_BOOL(float8_ge(arg1, (float8) arg2));
 }
 
-/*
- * Implements the float8 version of the width_bucket() function
- * defined by SQL2003. See also width_bucket_numeric().
- *
- * 'bound1' and 'bound2' are the lower and upper bounds of the
- * histogram's range, respectively. 'count' is the number of buckets
- * in the histogram. width_bucket() returns an integer indicating the
- * bucket number that 'operand' belongs to in an equiwidth histogram
- * with the specified characteristics. An operand smaller than the
- * lower bound is assigned to bucket 0. An operand greater than or equal
- * to the upper bound is assigned to an additional bucket (with number
- * count+1). We don't allow "NaN" for any of the float8 inputs, and we
- * don't allow either of the histogram bounds to be +/- infinity.
- */
 Datum
 width_bucket_float8(PG_FUNCTION_ARGS)
 {
-	float8		operand = PG_GETARG_FLOAT8(0);
-	float8		bound1 = PG_GETARG_FLOAT8(1);
-	float8		bound2 = PG_GETARG_FLOAT8(2);
-	int32		count = PG_GETARG_INT32(3);
-	int32		result;
-
-	if (count <= 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION),
-				 errmsg("count must be greater than zero")));
-
-	if (isnan(operand) || isnan(bound1) || isnan(bound2))
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION),
-				 errmsg("operand, lower bound, and upper bound cannot be NaN")));
-
-	/* Note that we allow "operand" to be infinite */
-	if (isinf(bound1) || isinf(bound2))
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION),
-				 errmsg("lower and upper bounds must be finite")));
-
-	if (bound1 < bound2)
-	{
-		if (operand < bound1)
-			result = 0;
-		else if (operand >= bound2)
-		{
-			if (pg_add_s32_overflow(count, 1, &result))
-				ereport(ERROR,
-						(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
-						 errmsg("integer out of range")));
-		}
-		else
-		{
-			if (!isinf(bound2 - bound1))
-			{
-				/* The quotient is surely in [0,1], so this can't overflow */
-				result = count * ((operand - bound1) / (bound2 - bound1));
-			}
-			else
-			{
-				/*
-				 * We get here if bound2 - bound1 overflows DBL_MAX.  Since
-				 * both bounds are finite, their difference can't exceed twice
-				 * DBL_MAX; so we can perform the computation without overflow
-				 * by dividing all the inputs by 2.  That should be exact too,
-				 * except in the case where a very small operand underflows to
-				 * zero, which would have negligible impact on the result
-				 * given such large bounds.
-				 */
-				result = count * ((operand / 2 - bound1 / 2) / (bound2 / 2 - bound1 / 2));
-			}
-			/* The quotient could round to 1.0, which would be a lie */
-			if (result >= count)
-				result = count - 1;
-			/* Having done that, we can add 1 without fear of overflow */
-			result++;
-		}
-	}
-	else if (bound1 > bound2)
-	{
-		if (operand > bound1)
-			result = 0;
-		else if (operand <= bound2)
-		{
-			if (pg_add_s32_overflow(count, 1, &result))
-				ereport(ERROR,
-						(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
-						 errmsg("integer out of range")));
-		}
-		else
-		{
-			if (!isinf(bound1 - bound2))
-				result = count * ((bound1 - operand) / (bound1 - bound2));
-			else
-				result = count * ((bound1 / 2 - operand / 2) / (bound1 / 2 - bound2 / 2));
-			if (result >= count)
-				result = count - 1;
-			result++;
-		}
-	}
-	else
-	{
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION),
-				 errmsg("lower bound cannot equal upper bound")));
-		result = 0;				/* keep the compiler quiet */
-	}
-
-	PG_RETURN_INT32(result);
+	elog(ERROR, "width_bucket_float8 is not implemented");
+	PG_RETURN_NULL();
 }
+
diff --git a/src/backend/utils/adt/numeric.c b/src/backend/utils/adt/numeric.c
index 14e8895..98f3683 100644
--- a/src/backend/utils/adt/numeric.c
+++ b/src/backend/utils/adt/numeric.c
@@ -1833,100 +1833,11 @@ generate_series_step_numeric(PG_FUNCTION_ARGS)
 }
 
 
-/*
- * Implements the numeric version of the width_bucket() function
- * defined by SQL2003. See also width_bucket_float8().
- *
- * 'bound1' and 'bound2' are the lower and upper bounds of the
- * histogram's range, respectively. 'count' is the number of buckets
- * in the histogram. width_bucket() returns an integer indicating the
- * bucket number that 'operand' belongs to in an equiwidth histogram
- * with the specified characteristics. An operand smaller than the
- * lower bound is assigned to bucket 0. An operand greater than or equal
- * to the upper bound is assigned to an additional bucket (with number
- * count+1). We don't allow "NaN" for any of the numeric inputs, and we
- * don't allow either of the histogram bounds to be +/- infinity.
- */
 Datum
 width_bucket_numeric(PG_FUNCTION_ARGS)
 {
-	Numeric		operand = PG_GETARG_NUMERIC(0);
-	Numeric		bound1 = PG_GETARG_NUMERIC(1);
-	Numeric		bound2 = PG_GETARG_NUMERIC(2);
-	int32		count = PG_GETARG_INT32(3);
-	NumericVar	count_var;
-	NumericVar	result_var;
-	int32		result;
-
-	if (count <= 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION),
-				 errmsg("count must be greater than zero")));
-
-	if (NUMERIC_IS_SPECIAL(operand) ||
-		NUMERIC_IS_SPECIAL(bound1) ||
-		NUMERIC_IS_SPECIAL(bound2))
-	{
-		if (NUMERIC_IS_NAN(operand) ||
-			NUMERIC_IS_NAN(bound1) ||
-			NUMERIC_IS_NAN(bound2))
-			ereport(ERROR,
-					(errcode(ERRCODE_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION),
-					 errmsg("operand, lower bound, and upper bound cannot be NaN")));
-		/* We allow "operand" to be infinite; cmp_numerics will cope */
-		if (NUMERIC_IS_INF(bound1) || NUMERIC_IS_INF(bound2))
-			ereport(ERROR,
-					(errcode(ERRCODE_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION),
-					 errmsg("lower and upper bounds must be finite")));
-	}
-
-	init_var(&result_var);
-	init_var(&count_var);
-
-	/* Convert 'count' to a numeric, for ease of use later */
-	int64_to_numericvar((int64) count, &count_var);
-
-	switch (cmp_numerics(bound1, bound2))
-	{
-		case 0:
-			ereport(ERROR,
-					(errcode(ERRCODE_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION),
-					 errmsg("lower bound cannot equal upper bound")));
-			break;
-
-			/* bound1 < bound2 */
-		case -1:
-			if (cmp_numerics(operand, bound1) < 0)
-				set_var_from_var(&const_zero, &result_var);
-			else if (cmp_numerics(operand, bound2) >= 0)
-				add_var(&count_var, &const_one, &result_var);
-			else
-				compute_bucket(operand, bound1, bound2, &count_var, false,
-							   &result_var);
-			break;
-
-			/* bound1 > bound2 */
-		case 1:
-			if (cmp_numerics(operand, bound1) > 0)
-				set_var_from_var(&const_zero, &result_var);
-			else if (cmp_numerics(operand, bound2) <= 0)
-				add_var(&count_var, &const_one, &result_var);
-			else
-				compute_bucket(operand, bound1, bound2, &count_var, true,
-							   &result_var);
-			break;
-	}
-
-	/* if result exceeds the range of a legal int4, we ereport here */
-	if (!numericvar_to_int32(&result_var, &result))
-		ereport(ERROR,
-				(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
-				 errmsg("integer out of range")));
-
-	free_var(&count_var);
-	free_var(&result_var);
-
-	PG_RETURN_INT32(result);
+	elog(ERROR, "width_bucket_numeric is not implemented");
+	PG_RETURN_NULL();
 }
 
 /*
